*** ./smali/com/android/server/power/ShutdownThread$2.smali	2014-10-11 10:21:18.692209667 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/services.jar.out/./smali/com/android/server/power/ShutdownThread$2.smali	2014-10-11 10:18:17.244205731 +1100
***************
*** 42,47 ****
--- 42,50 ----
      .parameter "which"
  
      .prologue
+     #calls: Lcom/android/server/power/ShutdownThread;->disableWakeAlarm()V
+     invoke-static {}, Lcom/android/server/power/ShutdownThread;->_disableWakeAlarm()V
+ 
      iget-object v0, p0, Lcom/android/server/power/ShutdownThread$2;->val$context:Landroid/content/Context;
  
      #calls: Lcom/android/server/power/ShutdownThread;->beginShutdownSequence(Landroid/content/Context;)V
