*** ./smali/com/android/internal/telephony/gsm/GsmDataConnectionTracker.smali	2014-10-11 10:20:04.380208055 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/GsmDataConnectionTracker.smali	2014-10-11 10:17:04.500204153 +1100
***************
*** 8,14 ****
      value = {
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$4;,
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ScriManager;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ApnChangeObserver;
      }
  .end annotation
  
--- 8,15 ----
      value = {
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$4;,
          Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ScriManager;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$ApnChangeObserver;,
!         Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;
      }
  .end annotation
  
***************
*** 5349,5354 ****
--- 5350,5364 ----
      goto :goto_0
  .end method
  
+ .method static getPolicyDataEnabled()Z
+     .locals 1
+ 
+     .prologue
+     sget-boolean v0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->sPolicyDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method private getPreferredApn()Lcom/android/internal/telephony/ApnSetting;
      .locals 12
  
***************
*** 5698,5704 ****
  
      if-eqz v0, :cond_0
  
!     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isDataAllowed()Z
  
      move-result v0
  
--- 5708,5714 ----
  
      if-eqz v0, :cond_0
  
!     invoke-static {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->isDataAllowed(Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;Lcom/android/internal/telephony/ApnContext;)Z
  
      move-result v0
  
***************
*** 9648,9653 ****
--- 9658,9697 ----
      return-void
  .end method
  
+ .method callApnIdToType(I)Ljava/lang/String;
+     .locals 1
+     .parameter "apnId"
+ 
+     .prologue
+     invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callIsDataAllowed()Z
+     .locals 1
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isDataAllowed()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method callIsMmsDataEnabled()Z
+     .locals 1
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->isMmsDataEnabled()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method protected cleanUpAllConnections(ZLjava/lang/String;)V
      .locals 4
      .parameter "tearDown"
***************
*** 10664,10678 ****
      :goto_0
      iput-boolean v4, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mUserDataEnabled:Z
  
!     iget-boolean v4, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mInternalDataEnabled:Z
  
!     if-eqz v4, :cond_0
! 
!     iget-boolean v4, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mUserDataEnabled:Z
! 
!     if-eqz v4, :cond_0
! 
!     sget-boolean v4, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->sPolicyDataEnabled:Z
  
      if-nez v4, :cond_2
  
--- 10708,10716 ----
      :goto_0
      iput-boolean v4, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mUserDataEnabled:Z
  
!     invoke-static {p0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->isAnyDataEnabled(Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;)Z
  
!     move-result v4
  
      if-nez v4, :cond_2
  
***************
*** 10860,10865 ****
--- 10898,10912 ----
      return-object v0
  .end method
  
+ .method getInternalDataEnabled()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mInternalDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method protected getLinkCapabilities(Ljava/lang/String;)Landroid/net/LinkCapabilities;
      .locals 4
      .parameter "apnType"
***************
*** 11238,11243 ****
--- 11285,11299 ----
      goto :goto_0
  .end method
  
+ .method getUserDataEnabled()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mUserDataEnabled:Z
+ 
+     return v0
+ .end method
+ 
  .method protected gotoIdleAndNotifyDataConnection(Ljava/lang/String;)V
      .locals 2
      .parameter "reason"
***************
*** 14451,14456 ****
--- 14507,14514 ----
      .end local v10           #proxy:Landroid/net/ProxyProperties;
      :cond_5
      :goto_3
+     invoke-static {v2}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->onDataSetupComplete(Lcom/android/internal/telephony/ApnContext;)V
+ 
      invoke-virtual {v2}, Lcom/android/internal/telephony/ApnContext;->getApnType()Ljava/lang/String;
  
      move-result-object v12
***************
*** 14940,14945 ****
--- 14998,15005 ----
  
      invoke-virtual {v0, v2}, Lcom/android/internal/telephony/ApnContext;->setState(Lcom/android/internal/telephony/DctConstants$State;)V
  
+     invoke-static {v0}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->onDisconnectDone(Lcom/android/internal/telephony/ApnContext;)V
+ 
      invoke-virtual {v0}, Lcom/android/internal/telephony/ApnContext;->isReactive()Z
  
      move-result v2
***************
*** 15140,15145 ****
--- 15200,15211 ----
      .prologue
      const/4 v1, 0x1
  
+     invoke-static {p0, p2, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker$Injector;->getApnType(Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;II)Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     iput-object v2, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
      iget-object v2, p0, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->mApnContexts:Ljava/util/concurrent/ConcurrentHashMap;
  
      invoke-virtual {p0, p1}, Lcom/android/internal/telephony/gsm/GsmDataConnectionTracker;->apnIdToType(I)Ljava/lang/String;
