*** ./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2014-10-11 10:20:03.892208044 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/GsmServiceStateTracker.smali	2014-10-11 10:17:04.148204146 +1100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$5;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$5;,
!         Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;
      }
  .end annotation
  
***************
*** 9885,9890 ****
--- 9886,9900 ----
      return v0
  .end method
  
+ .method getIccRecords()Lcom/android/internal/telephony/IccRecords;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mIccRecords:Lcom/android/internal/telephony/IccRecords;
+ 
+     return-object v0
+ .end method
+ 
  .method protected getPhone()Lcom/android/internal/telephony/Phone;
      .locals 1
  
***************
*** 9894,9899 ****
--- 9904,9918 ----
      return-object v0
  .end method
  
+ .method getRilRadioTechnology()I
+     .locals 1
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mRilRadioTechnology:I
+ 
+     return v0
+ .end method
+ 
  .method public handleMessage(Landroid/os/Message;)V
      .locals 37
      .parameter "msg"
***************
*** 14375,14383 ****
      invoke-virtual {v11}, Lcom/android/internal/telephony/gsm/SIMRecords;->getServiceProviderName()Ljava/lang/String;
  
      move-result-object v12
! 
      .local v12, spn:Ljava/lang/String;
      :goto_1
      const/4 v8, 0x0
  
      .local v8, sEons:Ljava/lang/String;
--- 14394,14408 ----
      invoke-virtual {v11}, Lcom/android/internal/telephony/gsm/SIMRecords;->getServiceProviderName()Ljava/lang/String;
  
      move-result-object v12
!     
      .local v12, spn:Ljava/lang/String;
      :goto_1
+     move-object/from16 v0, p0
+ 
+     invoke-static {v0, v12}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;->getSpn(Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v12
+     
      const/4 v8, 0x0
  
      .local v8, sEons:Ljava/lang/String;
***************
*** 14995,15007 ****
  
      move-object/from16 v0, p0
  
!     iget-object v14, v0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mServiceStateExt:Lcom/mediatek/common/telephony/IServiceStateExt;
! 
!     move-object/from16 v0, p0
! 
!     iget v15, v0, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;->mRilRadioTechnology:I
! 
!     invoke-interface {v14, v5, v15}, Lcom/mediatek/common/telephony/IServiceStateExt;->onUpdateSpnDisplay(Ljava/lang/String;I)Ljava/lang/String;
  
      move-result-object v5
  
--- 15020,15026 ----
  
      move-object/from16 v0, p0
  
!     invoke-static {v0, v5}, Lcom/android/internal/telephony/gsm/GsmServiceStateTracker$Injector;->getPlmn(Lcom/android/internal/telephony/gsm/GsmServiceStateTracker;Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v5
  
