*** ./smali/com/android/internal/telephony/gsm/SmsMessage.smali	2014-10-11 10:20:04.512208058 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/gsm/SmsMessage.smali	2014-10-11 10:17:04.568204155 +1100
***************
*** 3992,3997 ****
--- 3992,4001 ----
  
      move-result-object v3
  
+     iput-object v3, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->recipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
+     iget-object v3, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->recipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
      iput-object v3, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->destinationAddress:Lcom/android/internal/telephony/SmsAddress;
  
      iget-object v3, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->destinationAddress:Lcom/android/internal/telephony/SmsAddress;
***************
*** 4697,4702 ****
--- 4701,4729 ----
      return v0
  .end method
  
+ .method public getRecipientAddress()Ljava/lang/String;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->recipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/telephony/gsm/SmsMessage;->recipientAddress:Lcom/android/internal/telephony/gsm/GsmSmsAddress;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/gsm/GsmSmsAddress;->getAddressString()Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     :goto_0
+     return-object v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public getStatus()I
      .locals 1
  
