*** ./smali/android/content/pm/ResolveInfo.smali	2014-10-11 10:20:32.948208675 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/content/pm/ResolveInfo.smali	2014-10-11 10:17:29.156204688 +1100
***************
*** 9,15 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;
      }
  .end annotation
  
--- 9,16 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/content/pm/ResolveInfo$DisplayNameComparator;,
!         Landroid/content/pm/ResolveInfo$Injector;
      }
  .end annotation
  
***************
*** 717,722 ****
--- 718,727 ----
      move-result-object v2
  
      .local v2, dr:Landroid/graphics/drawable/Drawable;
+     invoke-static {v2}, Lmiui/content/res/IconCustomizer;->generateIconStyleDrawable(Landroid/graphics/drawable/Drawable;)Landroid/graphics/drawable/BitmapDrawable;
+ 
+     move-result-object v2
+ 
      if-eqz v2, :cond_1
  
      .end local v2           #dr:Landroid/graphics/drawable/Drawable;
***************
*** 744,750 ****
  
      iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     invoke-virtual {p1, v3, v4, v0}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 749,757 ----
  
      iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
  
!     iget-object v5, v1, Landroid/content/pm/PackageItemInfo;->name:Ljava/lang/String;
! 
!     invoke-static {p1, v3, v5, v4, v0}, Landroid/app/MiuiThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
