*** ./smali/com/android/internal/policy/impl/keyguard/KeyguardWidgetPager.smali	2014-10-11 10:20:59.892209259 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/keyguard/KeyguardWidgetPager.smali	2014-10-11 10:17:58.740205330 +1100
***************
*** 1474,1479 ****
--- 1474,1490 ----
  
      .local v0, isChallengeOverlapping:Z
      :goto_0
+     if-nez v0, :cond_miui_0
+ 
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/keyguard/KeyguardWidgetPager;->startReordering()Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_miui_0
+ 
+     const/4 v1, 0x1
+     
+     :cond_miui_0
      return v1
  
      .end local v0           #isChallengeOverlapping:Z
