*** ./smali/com/android/internal/policy/impl/KeyguardViewMediator$1.smali	2014-10-11 10:20:56.308209181 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/KeyguardViewMediator$1.smali	2014-10-11 10:17:55.132205252 +1100
***************
*** 374,379 ****
--- 374,598 ----
      .end packed-switch
  .end method
  
+ .method public onSimStateChangedGemini(Lcom/android/internal/telephony/IccCardConstants$State;I)V
+     .locals 5
+     .parameter "simState"
+     .parameter "simId"
+ 
+     .prologue
+     const-string v2, "KeyguardViewMediator"
+ 
+     new-instance v3, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v3}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v4, "onSimStateChangedGemini: "
+ 
+     invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v3
+ 
+     invoke-virtual {v3, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v3
+ 
+     const-string v4, ",simId = "
+ 
+     invoke-virtual {v3, v4}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v3
+ 
+     invoke-virtual {v3, p2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v3
+ 
+     invoke-virtual {v3}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v3
+ 
+     invoke-static {v2, v3}, Lcom/android/internal/policy/impl/keyguard/KeyguardUtils;->xlogD(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     sget-object v2, Lcom/android/internal/policy/impl/KeyguardViewMediator$5;->$SwitchMap$com$android$internal$telephony$IccCardConstants$State:[I
+ 
+     invoke-virtual {p1}, Lcom/android/internal/telephony/IccCardConstants$State;->ordinal()I
+ 
+     move-result v3
+ 
+     aget v2, v2, v3
+ 
+     packed-switch v2, :pswitch_data_0
+ 
+     :cond_0
+     :goto_0
+     :pswitch_0
+     return-void
+ 
+     :pswitch_1
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     #getter for: Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     invoke-static {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->access$800(Lcom/android/internal/policy/impl/KeyguardViewMediator;)Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, p2}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getLastSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v0
+ 
+     .local v0, lastSimState:Lcom/android/internal/telephony/IccCardConstants$State;
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-eq v0, v2, :cond_1
+ 
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne v0, v2, :cond_3
+ 
+     :cond_1
+     const/4 v1, 0x1
+ 
+     .local v1, lastWasLocked:Z
+     :goto_1
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     #getter for: Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     invoke-static {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->getUpdateMonitor(Lcom/android/internal/policy/impl/KeyguardViewMediator;)Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->isDeviceProvisioned()Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_2
+ 
+     if-eqz v1, :cond_0
+ 
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->ABSENT:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne p1, v2, :cond_0
+ 
+     :cond_2
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->isShowing()Z
+ 
+     move-result v2
+ 
+     if-nez v2, :cond_4
+ 
+     const-string v2, "KeyguardViewMediator"
+ 
+     const-string v3, "INTENT_VALUE_ICC_ABSENT and keygaurd isn\'t showing, we need to show the keyguard since the device isn\'t provisioned yet."
+ 
+     invoke-static {v2, v3}, Lcom/android/internal/policy/impl/keyguard/KeyguardUtils;->xlogD(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callDoKeyguardLocked()V
+ 
+     goto :goto_0
+ 
+     .end local v1           #lastWasLocked:Z
+     :cond_3
+     const/4 v1, 0x0
+ 
+     goto :goto_1
+ 
+     .restart local v1       #lastWasLocked:Z
+     :cond_4
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callResetStateLocked()V
+ 
+     goto :goto_0
+ 
+     .end local v0           #lastSimState:Lcom/android/internal/telephony/IccCardConstants$State;
+     .end local v1           #lastWasLocked:Z
+     :pswitch_2
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->isShowing()Z
+ 
+     move-result v2
+ 
+     if-nez v2, :cond_5
+ 
+     const-string v2, "KeyguardViewMediator"
+ 
+     const-string v3, "INTENT_VALUE_ICC_LOCKED and keygaurd isn\'t showing, we need to show the keyguard so the user can enter their sim pin"
+ 
+     invoke-static {v2, v3}, Lcom/android/internal/policy/impl/keyguard/KeyguardUtils;->xlogD(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callDoKeyguardLocked()V
+ 
+     goto :goto_0
+ 
+     :cond_5
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callResetStateLocked()V
+ 
+     goto :goto_0
+ 
+     :pswitch_3
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->isShowing()Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_0
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callResetStateLocked()V
+ 
+     goto :goto_0
+ 
+     :pswitch_4
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->isShowing()Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_0
+ 
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     iget-object v3, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     #getter for: Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     invoke-static {v3}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->getUpdateMonitor(Lcom/android/internal/policy/impl/KeyguardViewMediator;)Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     move-result-object v3
+ 
+     invoke-virtual {v3, p2}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getLastSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v3
+ 
+     if-ne v2, v3, :cond_0
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator$1;->this$0:Lcom/android/internal/policy/impl/KeyguardViewMediator;
+ 
+     invoke-virtual {v2}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->callResetStateLocked()V
+ 
+     goto/16 :goto_0
+ 
+     :pswitch_data_0
+     .packed-switch 0x1
+         :pswitch_4
+         :pswitch_1
+         :pswitch_2
+         :pswitch_2
+         :pswitch_0
+         :pswitch_3
+     .end packed-switch
+ .end method
+ 
  .method public onUserRemoved(I)V
      .locals 1
      .parameter "userId"
