*** ./smali/com/android/internal/policy/impl/KeyguardViewMediator.smali	2014-10-11 10:20:56.244209180 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/KeyguardViewMediator.smali	2014-10-11 10:17:55.064205250 +1100
***************
*** 121,126 ****
--- 121,128 ----
  
  .field private mPhoneState:Ljava/lang/String;
  
+ .field mRealPowerManager:Landroid/os/LocalPowerManager;
+ 
  .field private mScreenOn:Z
  
  .field private mShowKeyguardWakeLock:Landroid/os/PowerManager$WakeLock;
***************
*** 302,328 ****
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mAlarmManager:Landroid/app/AlarmManager;
  
!     new-instance v0, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     invoke-direct {v0, p1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;-><init>(Landroid/content/Context;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     new-instance v0, Lcom/android/internal/widget/LockPatternUtils;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mContext:Landroid/content/Context;
  
!     invoke-direct {v0, v1}, Lcom/android/internal/widget/LockPatternUtils;-><init>(Landroid/content/Context;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardViewProperties;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     invoke-direct {v0, v1, v3}, Lcom/android/internal/policy/impl/LockPatternKeyguardViewProperties;-><init>(Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewProperties:Lcom/android/internal/policy/impl/KeyguardViewProperties;
  
--- 304,330 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mAlarmManager:Landroid/app/AlarmManager;
  
!     new-instance v0, Lcom/android/internal/policy/impl/MiuiKeyguardUpdateMonitor;
  
!     invoke-direct {v0, p1}, Lcom/android/internal/policy/impl/MiuiKeyguardUpdateMonitor;-><init>(Landroid/content/Context;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     new-instance v0, Lmiui/security/MiuiLockPatternUtils;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mContext:Landroid/content/Context;
  
!     invoke-direct {v0, v1}, Lmiui/security/MiuiLockPatternUtils;-><init>(Landroid/content/Context;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     new-instance v0, Lcom/android/internal/policy/impl/MiuiLockPatternKeyguardViewProperties;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     invoke-direct {v0, v1, v3}, Lcom/android/internal/policy/impl/MiuiLockPatternKeyguardViewProperties;-><init>(Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewProperties:Lcom/android/internal/policy/impl/KeyguardViewProperties;
  
***************
*** 335,341 ****
      check-cast v2, Landroid/view/WindowManager;
  
      .local v2, wm:Landroid/view/WindowManager;
!     new-instance v0, Lcom/android/internal/policy/impl/KeyguardViewManager;
  
      iget-object v4, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewProperties:Lcom/android/internal/policy/impl/KeyguardViewProperties;
  
--- 337,343 ----
      check-cast v2, Landroid/view/WindowManager;
  
      .local v2, wm:Landroid/view/WindowManager;
!     new-instance v0, Lcom/android/internal/policy/impl/MiuiKeyguardViewManager;
  
      iget-object v4, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewProperties:Lcom/android/internal/policy/impl/KeyguardViewProperties;
  
***************
*** 345,351 ****
  
      move-object v3, p0
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/internal/policy/impl/KeyguardViewManager;-><init>(Landroid/content/Context;Landroid/view/ViewManager;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardViewProperties;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
  
--- 347,353 ----
  
      move-object v3, p0
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/internal/policy/impl/MiuiKeyguardViewManager;-><init>(Landroid/content/Context;Landroid/view/ViewManager;Lcom/android/internal/policy/impl/KeyguardViewCallback;Lcom/android/internal/policy/impl/KeyguardViewProperties;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
  
***************
*** 852,857 ****
--- 854,861 ----
      invoke-static {v2, v3}, Landroid/util/Log;->w(Ljava/lang/String;Ljava/lang/String;)I
  
      :goto_0
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->postAdjustStatusBarLocked()V
+ 
      return-void
  
      :cond_1
***************
*** 1179,1184 ****
--- 1183,1192 ----
  
      .local v0, lockedOrMissing:Z
      :goto_2
+     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->resetLockedOrMissing(ZZLcom/android/internal/telephony/IccCardConstants$State;)Z
+ 
+     move-result v0
+ 
      const-string v4, "KeyguardViewMediator"
  
      new-instance v5, Ljava/lang/StringBuilder;
***************
*** 1332,1338 ****
  
      const/4 v0, 0x0
  
!     invoke-direct {p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->playSounds(Z)V
  
      :cond_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
--- 1340,1346 ----
  
      const/4 v0, 0x0
  
!     invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->playSounds(Z)V
  
      :cond_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
***************
*** 1607,1612 ****
--- 1615,1633 ----
      return-void
  
      :cond_0
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-static {}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->isAlarmBoot()Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_1
+ 
+     const/4 v0, 0x1
+ 
+     invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->playSounds(Z)V
+ 
+     :goto_1
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
  
      invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardViewManager;->show()V
***************
*** 1635,1655 ****
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_0
  
-     :goto_1
-     :try_start_2
-     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
- 
-     invoke-static {}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->isAlarmBoot()Z
- 
-     move-result v0
- 
-     if-nez v0, :cond_1
- 
-     const/4 v0, 0x1
- 
-     invoke-direct {p0, v0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->playSounds(Z)V
- 
      :goto_2
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mShowKeyguardWakeLock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->release()V
--- 1656,1663 ----
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_0
  
      :goto_2
+     :try_start_2
      iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mShowKeyguardWakeLock:Landroid/os/PowerManager$WakeLock;
  
      invoke-virtual {v0}, Landroid/os/PowerManager$WakeLock;->release()V
***************
*** 1683,1694 ****
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
!     goto :goto_2
  
      :catch_0
      move-exception v0
  
!     goto :goto_1
  .end method
  
  .method private handleTimeout(I)V
--- 1691,1702 ----
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
  
!     goto :goto_1
  
      :catch_0
      move-exception v0
  
!     goto :goto_2
  .end method
  
  .method private handleTimeout(I)V
***************
*** 1966,1972 ****
      return-void
  .end method
  
! .method private playSounds(Z)V
      .locals 8
      .parameter "locked"
  
--- 1974,1980 ----
      return-void
  .end method
  
! .method protected playSounds(Z)V
      .locals 8
      .parameter "locked"
  
***************
*** 2293,2298 ****
--- 2301,2344 ----
  
  
  # virtual methods
+ .method callNotifyScreenOffLocked()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->notifyScreenOffLocked()V
+ 
+     return-void
+ .end method
+ 
+ .method callShowLocked()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->showLocked()V
+ 
+     return-void
+ .end method
+ 
+ .method public dismiss()V
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mShowing:Z
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mHidden:Z
+ 
+     if-nez v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mKeyguardViewManager:Lcom/android/internal/policy/impl/KeyguardViewManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardViewManager;->dismiss()V
+ 
+     :cond_0
+     return-void
+ .end method
+ 
  .method public doKeyguardTimeout()V
      .locals 3
  
***************
*** 2317,2322 ****
--- 2363,2411 ----
      return-void
  .end method
  
+ .method public doKeyguardTimeout(Landroid/os/Bundle;)V
+     .locals 3
+     .parameter "options"
+ 
+     .prologue
+     const/16 v2, 0xd
+ 
+     iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mHandler:Landroid/os/Handler;
+ 
+     invoke-virtual {v1, v2}, Landroid/os/Handler;->removeMessages(I)V
+ 
+     iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mHandler:Landroid/os/Handler;
+ 
+     invoke-virtual {v1, v2, p1}, Landroid/os/Handler;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
+ 
+     move-result-object v0
+ 
+     .local v0, msg:Landroid/os/Message;
+     iget-object v1, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mHandler:Landroid/os/Handler;
+ 
+     invoke-virtual {v1, v0}, Landroid/os/Handler;->sendMessage(Landroid/os/Message;)Z
+ 
+     return-void
+ .end method
+ 
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
+ .method getStatusBarManager()Landroid/app/StatusBarManager;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mStatusBarManager:Landroid/app/StatusBarManager;
+ 
+     return-object v0
+ .end method
+ 
  .method public isAlarmUnlockScreen()Z
      .locals 1
  
***************
*** 2340,2345 ****
--- 2429,2455 ----
      goto :goto_0
  .end method
  
+ .method public isDismissable()Z
+     .locals 1
+ 
+     .prologue
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->isSecure()Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_0
+ 
+     const/4 v0, 0x1
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public isInputRestricted()Z
      .locals 1
  
***************
*** 2557,2562 ****
--- 2667,2686 ----
      return-void
  .end method
  
+ .method public onDreamingStarted()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public onDreamingStopped()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
  .method public onScreenTurnedOff(I)V
      .locals 21
      .parameter "why"
***************
*** 3078,3083 ****
--- 3202,3245 ----
      throw v0
  .end method
  
+ .method public onWakeKeyWhenKeyguardShowingTq(I)V
+     .locals 3
+     .parameter "keyCode"
+ 
+     .prologue
+     const-string v0, "KeyguardViewMediator"
+ 
+     new-instance v1, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v2, "onWakeKeyWhenKeyguardShowing("
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     const-string v2, ")"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-static {v0, v1}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->wakeWhenReadyLocked(I)V
+ 
+     return-void
+ .end method
+ 
  .method public onWakeKeyWhenKeyguardShowingTq(IZ)Z
      .locals 3
      .parameter "keyCode"
***************
*** 3247,3252 ****
--- 3409,3433 ----
      throw v1
  .end method
  
+ .method postAdjustStatusBarLocked()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public setCurrentUser(I)V
+     .locals 1
+     .parameter "newUserId"
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     invoke-virtual {v0, p1}, Lcom/android/internal/widget/LockPatternUtils;->setCurrentUser(I)V
+ 
+     return-void
+ .end method
+ 
  .method public setHidden(Z)V
      .locals 6
      .parameter "isHidden"
***************
*** 3498,3503 ****
--- 3679,3731 ----
      goto :goto_1
  .end method
  
+ .method public showAssistant()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method suppressNextLockSound()V
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iput-boolean v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mSuppressNextLockSound:Z
+ 
+     return-void
+ .end method
+ 
+ .method public userActivity()V
+     .locals 2
+ 
+     .prologue
+     const-wide/16 v0, 0x2710
+ 
+     invoke-virtual {p0, v0, v1}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->userActivity(J)V
+ 
+     return-void
+ .end method
+ 
+ .method public userActivity(J)V
+     .locals 4
+     .parameter "holdMs"
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mPM:Landroid/os/PowerManager;
+ 
+     invoke-static {}, Landroid/os/SystemClock;->uptimeMillis()J
+ 
+     move-result-wide v1
+ 
+     const/4 v3, 0x0
+ 
+     invoke-virtual {v0, v1, v2, v3}, Landroid/os/PowerManager;->userActivity(JZ)V
+ 
+     return-void
+ .end method
+ 
  .method public verifyUnlock(Landroid/view/WindowManagerPolicy$OnKeyguardExitResult;)V
      .locals 2
      .parameter "callback"
***************
*** 3582,3584 ****
--- 3810,3931 ----
  
      goto :goto_0
  .end method
+ 
+ .method private resetLockedOrMissing(ZZLcom/android/internal/telephony/IccCardConstants$State;)Z
+     .locals 6
+     .parameter "lockedOrMissing"
+     .parameter "requireSim"
+     .parameter "state"
+ 
+     .prologue
+     const/4 v3, 0x0
+ 
+     const/4 v4, 0x1
+ 
+     invoke-static {}, Lcom/android/internal/policy/impl/keyguard/KeyguardUtils;->isGemini()Z
+ 
+     move-result v5
+ 
+     if-eqz v5, :cond_3
+ 
+     iget-object v5, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-virtual {v5, v4}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v2
+ 
+     .local v2, stateGemini:Lcom/android/internal/telephony/IccCardConstants$State;
+     invoke-virtual {p3}, Lcom/android/internal/telephony/IccCardConstants$State;->isPinLocked()Z
+ 
+     move-result v5
+ 
+     if-nez v5, :cond_0
+ 
+     sget-object v5, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne p3, v5, :cond_4
+ 
+     if-eqz p2, :cond_4
+ 
+     :cond_0
+     move v1, v4
+ 
+     .local v1, simLockedOrMissing:Z
+     :goto_0
+     invoke-virtual {v2}, Lcom/android/internal/telephony/IccCardConstants$State;->isPinLocked()Z
+ 
+     move-result v5
+ 
+     if-nez v5, :cond_1
+ 
+     sget-object v5, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne v2, v5, :cond_5
+ 
+     if-eqz p2, :cond_5
+ 
+     :cond_1
+     move v0, v4
+ 
+     .local v0, simGeminilockedOrMissing:Z
+     :goto_1
+     if-nez v1, :cond_2
+ 
+     if-eqz v0, :cond_6
+ 
+     :cond_2
+     move p1, v4
+ 
+     .end local v0           #simGeminilockedOrMissing:Z
+     .end local v1           #simLockedOrMissing:Z
+     .end local v2           #stateGemini:Lcom/android/internal/telephony/IccCardConstants$State;
+     :cond_3
+     :goto_2
+     return p1
+ 
+     .restart local v2       #stateGemini:Lcom/android/internal/telephony/IccCardConstants$State;
+     :cond_4
+     move v1, v3
+ 
+     goto :goto_0
+ 
+     .restart local v1       #simLockedOrMissing:Z
+     :cond_5
+     move v0, v3
+ 
+     goto :goto_1
+ 
+     .restart local v0       #simGeminilockedOrMissing:Z
+     :cond_6
+     move p1, v3
+ 
+     goto :goto_2
+ .end method
+ 
+ .method callDoKeyguardLocked()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->doKeyguardLocked()V
+ 
+     return-void
+ .end method
+ 
+ .method callResetStateLocked()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/KeyguardViewMediator;->resetStateLocked()V
+ 
+     return-void
+ .end method
+ 
+ .method static synthetic getUpdateMonitor(Lcom/android/internal/policy/impl/KeyguardViewMediator;)Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     .locals 1
+     .parameter "x0"
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/KeyguardViewMediator;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     return-object v0
+ .end method
