*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardViewProperties.smali	2014-10-11 10:20:56.160209178 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardViewProperties.smali	2014-10-11 10:17:54.976205248 +1100
***************
*** 29,64 ****
  .end method
  
  .method private isSimPinSecure()Z
!     .locals 2
  
      .prologue
!     iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardViewProperties;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
!     invoke-virtual {v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
  
      move-result-object v0
  
!     .local v0, simState:Lcom/android/internal/telephony/IccCardConstants$State;
!     sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v0, v1, :cond_0
  
!     sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v0, v1, :cond_0
  
!     sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-ne v0, v1, :cond_1
  
!     :cond_0
!     const/4 v1, 0x1
  
      :goto_0
!     return v1
  
      :cond_1
!     const/4 v1, 0x0
  
      goto :goto_0
  .end method
--- 29,83 ----
  .end method
  
  .method private isSimPinSecure()Z
!     .locals 4
  
      .prologue
!     const/4 v2, 0x1
  
!     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardViewProperties;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
! 
!     invoke-virtual {v3}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     move-result-object v1
! 
!     .local v1, simState:Lcom/android/internal/telephony/IccCardConstants$State;
!     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardViewProperties;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
! 
!     invoke-virtual {v3, v2}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
  
      move-result-object v0
  
!     .local v0, sim2State:Lcom/android/internal/telephony/IccCardConstants$State;
!     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v1, v3, :cond_0
  
!     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v1, v3, :cond_0
  
!     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v1, v3, :cond_0
  
!     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     if-eq v0, v3, :cond_0
  
+     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-eq v0, v3, :cond_0
+ 
+     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne v0, v3, :cond_1
+ 
+     :cond_0
      :goto_0
!     return v2
  
      :cond_1
!     const/4 v2, 0x0
  
      goto :goto_0
  .end method
