*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView$3.smali	2014-10-11 10:21:00.568209274 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView$3.smali	2014-10-11 10:17:59.424205345 +1100
***************
*** 7,14 ****
  
  
  # annotations
! .annotation system Ldalvik/annotation/EnclosingClass;
!     value = Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
--- 7,14 ----
  
  
  # annotations
! .annotation system Ldalvik/annotation/EnclosingMethod;
!     value = Lcom/android/internal/policy/impl/LockPatternKeyguardView;->createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
***************
*** 130,136 ****
  .end method
  
  .method public goToUnlockScreen()V
!     .locals 4
  
      .prologue
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
--- 130,136 ----
  .end method
  
  .method public goToUnlockScreen()V
!     .locals 5
  
      .prologue
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
***************
*** 147,152 ****
--- 147,166 ----
      .local v0, simState:Lcom/android/internal/telephony/IccCardConstants$State;
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
+     #getter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     invoke-static {v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$400(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     move-result-object v1
+     
+     const/4 v2, 0x1
+     
+     invoke-virtual {v1, v2}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v4
+ 
+     .local v4, sim2State:Lcom/android/internal/telephony/IccCardConstants$State;
+     iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
+ 
      #calls: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
      invoke-static {v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->access$700(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)Z
  
***************
*** 156,163 ****
  
      sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-ne v0, v1, :cond_1
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      #getter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
--- 170,179 ----
  
      sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v0, v1, :cond_miui_0
  
+     if-ne v4, v1, :cond_1
+     :cond_miui_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
      #getter for: Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
***************
*** 392,398 ****
      move-result v2
  
      .local v2, failedAttemptsBeforeWipe:I
!     const/16 v0, 0xf
  
      .local v0, failedAttemptWarning:I
      if-lez v2, :cond_2
--- 408,414 ----
      move-result v2
  
      .local v2, failedAttemptsBeforeWipe:I
!     const/16 v0, 0x8
  
      .local v0, failedAttemptWarning:I
      if-lez v2, :cond_2
***************
*** 466,472 ****
  
      .local v4, showTimeout:Z
      :goto_3
!     if-eqz v5, :cond_5
  
      iget-object v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
--- 482,490 ----
  
      .local v4, showTimeout:Z
      :goto_3
!     sget-boolean v8, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->FALSE:Z
! 
!     if-nez v8, :cond_5
  
      iget-object v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;->this$0:Lcom/android/internal/policy/impl/LockPatternKeyguardView;
  
***************
*** 507,513 ****
  
      .restart local v4       #showTimeout:Z
      :cond_7
!     const/16 v8, 0x14
  
      if-lt v1, v8, :cond_5
  
--- 525,531 ----
  
      .restart local v4       #showTimeout:Z
      :cond_7
!     const/16 v8, 0x9
  
      if-lt v1, v8, :cond_5
  
