*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2014-10-11 10:20:56.544209186 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2014-10-11 10:17:55.384205257 +1100
***************
*** 10,16 ****
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$FastBitmapDrawable;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
      }
  .end annotation
  
--- 10,17 ----
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$FastBitmapDrawable;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$AccountAnalyzer;,
          Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;,
!         Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;
      }
  .end annotation
  
***************
*** 70,82 ****
  
  .field private mShowLockBeforeUnlock:Z
  
! .field private mSuppressBiometricUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
  .field private mUnlockScreen:Landroid/view/View;
  
! .field private mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
  .field private final mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
--- 71,85 ----
  
  .field private mShowLockBeforeUnlock:Z
  
! .field private mSimIdToUnlock:I
! 
! .field protected mSuppressBiometricUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
  .field private mUnlockScreen:Landroid/view/View;
  
! .field protected mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
  .field private final mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
***************
*** 140,145 ****
--- 143,152 ----
  
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mIsVerifyUnlockOnly:Z
  
+     const/4 v0, -0x1
+ 
+     iput v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
+ 
      new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$1;
  
      invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$1;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
***************
*** 152,160 ****
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mWidgetCallback:Lcom/android/internal/widget/LockScreenWidgetCallback;
  
!     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
  
!     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
--- 159,167 ----
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mWidgetCallback:Lcom/android/internal/widget/LockScreenWidgetCallback;
  
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
!     move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
***************
*** 200,205 ****
--- 207,224 ----
  
      sput-boolean v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->sIsFirstAppearanceAfterBoot:Z
  
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getBatteryStatus()Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$BatteryStatus;
+ 
+     move-result-object v0
+ 
+     invoke-virtual {v0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$BatteryStatus;->isCharged()Z
+ 
+     move-result v0
+ 
+     iput-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mPluggedIn:Z
+ 
      const-string v0, "power"
  
      invoke-virtual {p1, v0}, Landroid/content/Context;->getSystemService(Ljava/lang/String;)Ljava/lang/Object;
***************
*** 343,349 ****
      .parameter "x0"
  
      .prologue
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showAlmostAtAccountLoginDialog()V
  
      return-void
  .end method
--- 362,368 ----
      .parameter "x0"
  
      .prologue
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showAlmostAtAccountLoginDialog()V
  
      return-void
  .end method
***************
*** 353,359 ****
      .parameter "x0"
  
      .prologue
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showTimeoutDialog()V
  
      return-void
  .end method
--- 372,378 ----
      .parameter "x0"
  
      .prologue
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showTimeoutDialog()V
  
      return-void
  .end method
***************
*** 509,517 ****
  .end method
  
  .method private getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
!     .locals 2
  
      .prologue
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
--- 528,538 ----
  .end method
  
  .method private getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
!     .locals 4
  
      .prologue
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->updateShowLockBeforeUnlock(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
***************
*** 519,524 ****
--- 540,554 ----
      move-result-object v0
  
      .local v0, simState:Lcom/android/internal/telephony/IccCardConstants$State;
+     iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     const/4 v3, 0x1
+ 
+     invoke-virtual {v1, v3}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v2
+ 
+     .local v2, sim2State:Lcom/android/internal/telephony/IccCardConstants$State;
      invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
  
      move-result v1
***************
*** 527,534 ****
  
      sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-ne v0, v1, :cond_1
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      invoke-virtual {v1}, Lcom/android/internal/widget/LockPatternUtils;->isPukUnlockScreenEnable()Z
--- 557,567 ----
  
      sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
!     if-eq v0, v1, :cond_miui_0
  
+     if-ne v2, v1, :cond_1
+     
+     :cond_miui_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      invoke-virtual {v1}, Lcom/android/internal/widget/LockPatternUtils;->isPukUnlockScreenEnable()Z
***************
*** 569,574 ****
--- 602,610 ----
      .locals 6
  
      .prologue
+     const/4 v5, 0x1
+ 
+     const/4 v4, 0x0
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
  
      invoke-virtual {v3}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
***************
*** 576,581 ****
--- 612,624 ----
      move-result-object v2
  
      .local v2, simState:Lcom/android/internal/telephony/IccCardConstants$State;
+     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-virtual {v3, v5}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v1
+ 
+     .local v1, sim2State:Lcom/android/internal/telephony/IccCardConstants$State;
      sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
  
      if-ne v2, v3, :cond_0
***************
*** 583,588 ****
--- 626,633 ----
      sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPin:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .local v0, currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     iput v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
+ 
      :goto_0
      return-object v0
  
***************
*** 595,604 ****
--- 640,677 ----
      sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPuk:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     iput v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
+ 
      goto :goto_0
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :cond_1
+     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne v1, v3, :cond_miui_0
+ 
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPin:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     iput v5, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
+ 
+     goto :goto_0
+ 
+     .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     :cond_miui_0
+     sget-object v3, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-ne v1, v3, :cond_miui_1
+ 
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPuk:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     iput v5, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
+ 
+     goto :goto_0
+ 
+     .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     :cond_miui_1
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
      invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->getKeyguardStoredPasswordQuality()I
***************
*** 633,653 ****
      throw v3
  
      :sswitch_0
!     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Password:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      goto :goto_0
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :sswitch_1
-     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
- 
-     invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->isLockPatternEnabled()Z
- 
-     move-result v3
- 
-     if-eqz v3, :cond_4
- 
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
  
      if-nez v3, :cond_2
--- 706,720 ----
      throw v3
  
      :sswitch_0
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockModeForHighPasswordQuality()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     move-result-object v0
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      goto :goto_0
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :sswitch_1
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
  
      if-nez v3, :cond_2
***************
*** 668,673 ****
--- 735,747 ----
  
      .end local v0           #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
      :cond_3
+     iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     invoke-virtual {v3}, Lcom/android/internal/widget/LockPatternUtils;->isLockPatternEnabled()Z
+ 
+     move-result v3
+ 
+     if-eqz v3, :cond_4
      sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Pattern:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
      .restart local v0       #currentMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
***************
*** 691,696 ****
--- 765,798 ----
      .end sparse-switch
  .end method
  
+ .method private getUnlockModeForHighPasswordQuality()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
+ 
+     if-nez v0, :cond_0
+ 
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/widget/LockPatternUtils;->isPermanentlyLocked()Z
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_1
+ 
+     :cond_0
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Account:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     :goto_0
+     return-object v0
+ 
+     :cond_1
+     sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->Password:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     goto :goto_0
+ .end method
+ 
  .method private initializeBiometricUnlockView(Landroid/view/View;)V
      .locals 7
      .parameter "view"
***************
*** 743,751 ****
  
      if-eqz v2, :cond_3
  
!     const v2, 0x10202ad
! 
!     invoke-virtual {p1, v2}, Landroid/view/View;->findViewById(I)Landroid/view/View;
  
      move-result-object v0
  
--- 845,851 ----
  
      if-eqz v2, :cond_3
  
!     invoke-virtual {p0, p1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getFaceLockAreaView(Landroid/view/View;)Landroid/view/View;
  
      move-result-object v0
  
***************
*** 916,962 ****
      return v0
  
      :pswitch_2
!     iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
! 
!     invoke-virtual {v4}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     move-result-object v4
! 
!     sget-object v5, Lcom/android/internal/telephony/IccCardConstants$State;->PIN_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     if-ne v4, v5, :cond_0
! 
!     move v0, v2
  
-     :goto_1
      goto :goto_0
  
-     :cond_0
-     move v0, v3
- 
-     goto :goto_1
- 
      :pswitch_3
!     iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
! 
!     invoke-virtual {v4}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState()Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     move-result-object v4
! 
!     sget-object v5, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
! 
!     if-ne v4, v5, :cond_1
! 
!     move v0, v2
  
-     :goto_2
      goto :goto_0
  
-     :cond_1
-     move v0, v3
- 
-     goto :goto_2
- 
      :pswitch_4
      const/4 v0, 0x1
  
--- 1016,1030 ----
      return v0
  
      :pswitch_2
!     const/4 v0, 0x1
  
      goto :goto_0
  
      :pswitch_3
!     const/4 v0, 0x1
  
      goto :goto_0
  
      :pswitch_4
      const/4 v0, 0x1
  
***************
*** 1009,1014 ****
--- 1077,1093 ----
      .locals 3
  
      .prologue
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     move-result-object v1
+ 
+     sget-object v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->LockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     if-ne v1, v2, :cond_miui_0
+     
+     return-void
+ 
+     :cond_miui_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mBiometricUnlock:Lcom/android/internal/policy/impl/BiometricSensorUnlock;
  
      if-eqz v1, :cond_0
***************
*** 1076,1081 ****
--- 1155,1169 ----
      .locals 2
  
      .prologue
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->needRecreateLockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)Z
+ 
+     move-result v0
+ 
+     if-nez v0, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
  
      if-eqz v0, :cond_0
***************
*** 1203,1216 ****
      return-void
  .end method
  
! .method private showAlmostAtAccountLoginDialog()V
      .locals 8
  
      .prologue
      const/16 v2, 0x1e
  
      .local v2, timeoutInSeconds:I
!     const/16 v0, 0xf
  
      .local v0, count:I
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 1291,1304 ----
      return-void
  .end method
  
! .method protected showAlmostAtAccountLoginDialog()V
      .locals 8
  
      .prologue
      const/16 v2, 0x1e
  
      .local v2, timeoutInSeconds:I
!     const/16 v0, 0x4
  
      .local v0, count:I
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1258,1264 ****
      .local v1, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-direct {p0, v3, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
--- 1346,1352 ----
      .local v1, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-virtual {p0, v3, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
***************
*** 1303,1314 ****
      .local v0, message:Ljava/lang/String;
      const/4 v2, 0x0
  
!     invoke-direct {p0, v2, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
! .method private showDialog(Ljava/lang/String;Ljava/lang/String;)V
      .locals 4
      .parameter "title"
      .parameter "message"
--- 1391,1402 ----
      .local v0, message:Ljava/lang/String;
      const/4 v2, 0x0
  
!     invoke-virtual {p0, v2, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
! .method protected showDialog(Ljava/lang/String;Ljava/lang/String;)V
      .locals 4
      .parameter "title"
      .parameter "message"
***************
*** 1358,1364 ****
      return-void
  .end method
  
! .method private showTimeoutDialog()V
      .locals 7
  
      .prologue
--- 1446,1452 ----
      return-void
  .end method
  
! .method protected showTimeoutDialog()V
      .locals 7
  
      .prologue
***************
*** 1425,1431 ****
      .local v0, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-direct {p0, v3, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  
--- 1513,1519 ----
      .local v0, message:Ljava/lang/String;
      const/4 v3, 0x0
  
!     invoke-virtual {p0, v3, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  
***************
*** 1464,1474 ****
      .local v0, message:Ljava/lang/String;
      const/4 v1, 0x0
  
!     invoke-direct {p0, v1, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
  .method private stuckOnLockScreenBecauseSimMissing()Z
      .locals 2
  
--- 1552,1608 ----
      .local v0, message:Ljava/lang/String;
      const/4 v1, 0x0
  
!     invoke-virtual {p0, v1, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->showDialog(Ljava/lang/String;Ljava/lang/String;)V
  
      return-void
  .end method
  
+ .method private skipUpdateSimUnlockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;)Z
+     .locals 4
+     .parameter "mode"
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     move-result-object v1
+ 
+     .local v1, unlockMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     const/4 v0, 0x0
+ 
+     .local v0, res:Z
+     sget-object v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->UnlockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     if-ne p1, v2, :cond_2
+ 
+     sget-object v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPin:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     if-ne v1, v2, :cond_0
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     sget-object v3, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPin:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     if-eq v2, v3, :cond_1
+ 
+     :cond_0
+     sget-object v2, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPuk:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     if-ne v1, v2, :cond_2
+ 
+     iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     sget-object v3, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPuk:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     if-ne v2, v3, :cond_2
+ 
+     :cond_1
+     const/4 v0, 0x1
+ 
+     .end local v0           #res:Z
+     :cond_2
+     return v0
+ .end method
+ 
  .method private stuckOnLockScreenBecauseSimMissing()Z
      .locals 2
  
***************
*** 1507,1513 ****
--- 1641,1667 ----
  
      :cond_0
      const/4 v0, 0x1
+     
+     iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-virtual {v1, v0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v1
+ 
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->ABSENT:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-eq v1, v2, :cond_miui_0
+ 
+     iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     invoke-virtual {v1, v0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v1
+ 
+     sget-object v2, Lcom/android/internal/telephony/IccCardConstants$State;->PERM_DISABLED:Lcom/android/internal/telephony/IccCardConstants$State;
  
+     if-ne v1, v2, :cond_1
+     :cond_miui_0
      :goto_0
      return v0
  
***************
*** 1565,1571 ****
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
!     if-eq v1, v3, :cond_4
  
      :cond_3
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->recreateUnlockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;)V
--- 1719,1741 ----
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUnlockScreenMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
  
!     if-ne v1, v3, :cond_3
! 
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     move-result-object v3
! 
!     sget-object v4, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPin:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     if-eq v3, v4, :cond_3
! 
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     move-result-object v3
! 
!     sget-object v4, Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;->SimPuk:Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
! 
!     if-ne v3, v4, :cond_4
  
      :cond_3
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->recreateUnlockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;)V
***************
*** 1697,1702 ****
--- 1867,1874 ----
  
      .restart local v2       #visibleScreen:Landroid/view/View;
      :cond_a
+     invoke-static {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Injector;->startBiometricUnlock(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
      return-void
  .end method
  
***************
*** 1780,1785 ****
--- 1952,1999 ----
  
  
  # virtual methods
+ .method callGetInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+     .locals 1
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callGetUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+     .locals 1
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getUnlockMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$UnlockMode;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
+ .method callMaybeStartBiometricUnlock()V
+     .locals 0
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->maybeStartBiometricUnlock()V
+ 
+     return-void
+ .end method
+ 
+ .method callStuckOnLockScreenBecauseSimMissing()Z
+     .locals 1
+ 
+     .prologue
+     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public cleanUp()V
      .locals 2
  
***************
*** 1848,1853 ****
--- 2062,2078 ----
      return-void
  .end method
  
+ .method protected createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
+     .locals 1
+ 
+     .prologue
+     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
+ 
+     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
+     return-object v0
+ .end method
+ 
  .method createLockScreen()Landroid/view/View;
      .locals 6
  
***************
*** 1905,1910 ****
--- 2130,2137 ----
      .parameter "unlockMode"
  
      .prologue
+     const/4 v8, 0x1
+ 
      const/4 v1, 0x0
  
      .local v1, unlockView:Landroid/view/View;
***************
*** 1912,1918 ****
  
      if-ne p1, v3, :cond_0
  
!     new-instance v0, Lcom/android/internal/policy/impl/PatternUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
--- 2139,2145 ----
  
      if-ne p1, v3, :cond_0
  
!     new-instance v0, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
  
      iget-object v1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
***************
*** 1931,1946 ****
  
      move-result v6
  
!     invoke-direct/range {v0 .. v6}, Lcom/android/internal/policy/impl/PatternUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;I)V
  
!     .local v0, view:Lcom/android/internal/policy/impl/PatternUnlockScreen;
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     invoke-virtual {v0, v3}, Lcom/android/internal/policy/impl/PatternUnlockScreen;->setEnableFallback(Z)V
  
      move-object v1, v0
  
!     .end local v0           #view:Lcom/android/internal/policy/impl/PatternUnlockScreen;
      .restart local v1       #unlockView:Landroid/view/View;
      :goto_0
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->initializeTransportControlView(Landroid/view/View;)V
--- 2158,2173 ----
  
      move-result v6
  
!     invoke-direct/range {v0 .. v6}, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;I)V
  
!     .local v0, view:Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
      iget-boolean v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
  
!     invoke-virtual {v0, v3}, Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;->setEnableFallback(Z)V
  
      move-object v1, v0
  
!     .end local v0           #view:Lcom/android/internal/policy/impl/MiuiPatternUnlockScreen;
      .restart local v1       #unlockView:Landroid/view/View;
      :goto_0
      invoke-direct {p0, v1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->initializeTransportControlView(Landroid/view/View;)V
***************
*** 1959,1965 ****
  
      if-ne p1, v3, :cond_1
  
!     new-instance v1, Lcom/android/internal/policy/impl/SimPukUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2186,2192 ----
  
      if-ne p1, v3, :cond_1
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiSimPUKUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1972,1978 ****
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/SimPukUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
--- 2199,2207 ----
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
! 
!     invoke-direct/range {v1 .. v8}, Lcom/android/internal/policy/impl/MiuiSimPUKUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;IZ)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
***************
*** 1982,1988 ****
  
      if-ne p1, v3, :cond_2
  
!     new-instance v1, Lcom/android/internal/policy/impl/SimUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2211,2217 ----
  
      if-ne p1, v3, :cond_2
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiSimPINUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v2, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 1995,2001 ****
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v1 .. v6}, Lcom/android/internal/policy/impl/SimUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
--- 2224,2232 ----
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mSimIdToUnlock:I
! 
!     invoke-direct/range {v1 .. v8}, Lcom/android/internal/policy/impl/MiuiSimPINUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;IZ)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
***************
*** 2006,2012 ****
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/AccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
--- 2237,2243 ----
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
***************
*** 2018,2024 ****
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v2 .. v7}, Lcom/android/internal/policy/impl/AccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
--- 2249,2257 ----
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     iget-boolean v8, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mForgotPattern:Z
! 
!     invoke-direct/range {v2 .. v8}, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;Z)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 2054,2060 ****
  
      if-ne p1, v3, :cond_4
  
!     new-instance v1, Lcom/android/internal/policy/impl/PasswordUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
--- 2287,2293 ----
  
      if-ne p1, v3, :cond_4
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;
  
      .end local v1           #unlockView:Landroid/view/View;
      iget-object v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
***************
*** 2069,2077 ****
  
      move-object v3, v1
  
!     invoke-direct/range {v3 .. v8}, Lcom/android/internal/policy/impl/PasswordUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
      goto :goto_0
  
      :cond_4
--- 2302,2318 ----
  
      move-object v3, v1
  
!     invoke-direct/range {v3 .. v8}, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/widget/LockPatternUtils;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;)V
  
      .restart local v1       #unlockView:Landroid/view/View;
+     move-object v3, v1
+ 
+     check-cast v3, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;
+ 
+     iget-boolean v4, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mEnableFallback:Z
+ 
+     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/MiuiPasswordUnlockScreen;->setEnableFallback(Z)V
+ 
      goto :goto_0
  
      :cond_4
***************
*** 2148,2153 ****
--- 2389,2467 ----
      return v1
  .end method
  
+ .method getConfiguration()Landroid/content/res/Configuration;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mConfiguration:Landroid/content/res/Configuration;
+ 
+     return-object v0
+ .end method
+ 
+ .method protected getFaceLockAreaView(Landroid/view/View;)Landroid/view/View;
+     .locals 1
+     .parameter "view"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
+ .method getLockPatternUtils()Lcom/android/internal/widget/LockPatternUtils;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     return-object v0
+ .end method
+ 
+ .method getLockScreen()Landroid/view/View;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
+ 
+     return-object v0
+ .end method
+ 
+ .method getMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     return-object v0
+ .end method
+ 
+ .method getScreenOn()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mScreenOn:Z
+ 
+     return v0
+ .end method
+ 
+ .method getShowLockBeforeUnlock()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
+ 
+     return v0
+ .end method
+ 
+ .method getUpdateMonitor()Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     return-object v0
+ .end method
+ 
  .method protected onConfigurationChanged(Landroid/content/res/Configuration;)V
      .locals 2
      .parameter "newConfig"
***************
*** 2398,2405 ****
      return-void
  .end method
  
  .method public show()V
!     .locals 1
  
      .prologue
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
--- 2712,2749 ----
      return-void
  .end method
  
+ .method setMode(Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     return-void
+ .end method
+ 
+ .method setScreenOn(Z)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mScreenOn:Z
+ 
+     return-void
+ .end method
+ 
+ .method setShowLockBeforeUnlock(Z)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-boolean p1, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mShowLockBeforeUnlock:Z
+ 
+     return-void
+ .end method
+ 
  .method public show()V
!     .locals 2
  
      .prologue
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockScreen:Landroid/view/View;
***************
*** 2417,2422 ****
--- 2761,2772 ----
  
      if-eqz v0, :cond_1
  
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     sget-object v1, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->UnlockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
+ 
+     if-ne v0, v1, :cond_1
+ 
      iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUnlockScreen:Landroid/view/View;
  
      check-cast v0, Lcom/android/internal/policy/impl/KeyguardScreen;
***************
*** 2525,2530 ****
--- 2875,2892 ----
  
      if-eq v0, v1, :cond_0
  
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     const/4 v1, 0x1
+ 
+     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->getSimState(I)Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     move-result-object v0
+ 
+     sget-object v1, Lcom/android/internal/telephony/IccCardConstants$State;->PUK_REQUIRED:Lcom/android/internal/telephony/IccCardConstants$State;
+ 
+     if-eq v0, v1, :cond_0
+ 
      sget-object v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->UnlockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
  
      const/4 v1, 0x0
***************
*** 2549,2551 ****
--- 2911,2926 ----
  
      goto :goto_0
  .end method
+ 
+ .method static synthetic _skipUpdateSimUnlockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView;Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;)Z
+     .locals 1
+     .parameter "x0"
+     .parameter "x1"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->skipUpdateSimUnlockScreen(Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;)Z
+ 
+     move-result v0
+ 
+     return v0
+ .end method
\ No newline at end of file
