*** ./smali/com/android/internal/policy/impl/PhoneWindowManager.smali	2014-10-11 10:21:01.216209288 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindowManager.smali	2014-10-11 10:18:00.124205360 +1100
***************
*** 14,20 ****
          Lcom/android/internal/policy/impl/PhoneWindowManager$MyOrientationListener;,
          Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;,
          Lcom/android/internal/policy/impl/PhoneWindowManager$PolicyHandler;,
!         Lcom/android/internal/policy/impl/PhoneWindowManager$PointerLocationInputEventReceiver;
      }
  .end annotation
  
--- 14,21 ----
          Lcom/android/internal/policy/impl/PhoneWindowManager$MyOrientationListener;,
          Lcom/android/internal/policy/impl/PhoneWindowManager$SettingsObserver;,
          Lcom/android/internal/policy/impl/PhoneWindowManager$PolicyHandler;,
!         Lcom/android/internal/policy/impl/PhoneWindowManager$PointerLocationInputEventReceiver;,
!         Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;
      }
  .end annotation
  
***************
*** 139,144 ****
--- 140,147 ----
  
  
  # instance fields
+ .field mAboveStatusBarFullScreenWindow:Landroid/view/WindowManagerPolicy$WindowState;
+ 
  .field mAccelerometerDefault:Z
  
  .field mAllowAllRotations:I
***************
*** 244,250 ****
  
  .field private mFreezeKey:Z
  
! .field mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
  .field private mHDMIObserver:Landroid/os/UEventObserver;
  
--- 247,253 ----
  
  .field private mFreezeKey:Z
  
! .field mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
  .field private mHDMIObserver:Landroid/os/UEventObserver;
  
***************
*** 294,300 ****
  
  .field mKeyguard:Landroid/view/WindowManagerPolicy$WindowState;
  
! .field mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
  .field mLandscapeRotation:I
  
--- 297,303 ----
  
  .field mKeyguard:Landroid/view/WindowManagerPolicy$WindowState;
  
! .field mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
  .field mLandscapeRotation:I
  
***************
*** 366,372 ****
  
  .field private mPowerKeyTriggered:Z
  
! .field private final mPowerLongPress:Ljava/lang/Runnable;
  
  .field mPowerManager:Landroid/os/PowerManager;
  
--- 369,375 ----
  
  .field private mPowerKeyTriggered:Z
  
! .field private mPowerLongPress:Ljava/lang/Runnable;
  
  .field mPowerManager:Landroid/os/PowerManager;
  
***************
*** 442,447 ****
--- 445,452 ----
  
  .field mStatusBar:Landroid/view/WindowManagerPolicy$WindowState;
  
+ .field mStatusBarDisableToken:Landroid/os/IBinder;
+ 
  .field mStatusBarHeight:I
  
  .field mStatusBarLayer:I
***************
*** 635,640 ****
--- 640,651 ----
  
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Landroid/os/Binder;
+ 
+     invoke-direct {v0}, Landroid/os/Binder;-><init>()V
+ 
+     iput-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarDisableToken:Landroid/os/IBinder;
+ 
      new-instance v0, Ljava/lang/Object;
  
      invoke-direct {v0}, Ljava/lang/Object;-><init>()V
***************
*** 1438,1446 ****
      .locals 3
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowing()Z
  
      move-result v0
  
--- 1449,1457 ----
      .locals 3
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowing()Z
  
      move-result v0
  
***************
*** 1977,1983 ****
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-nez v0, :cond_0
  
--- 1988,1994 ----
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-nez v0, :cond_0
  
***************
*** 1987,1995 ****
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v0
  
--- 1998,2006 ----
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v0
  
***************
*** 2517,2529 ****
  
      if-eqz v1, :cond_2
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v1, :cond_2
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isSecure()Z
  
      move-result v1
  
--- 2528,2540 ----
  
      if-eqz v1, :cond_2
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v1, :cond_2
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isSecure()Z
  
      move-result v1
  
***************
*** 2693,2699 ****
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mTopFullscreenOpaqueWindowState:Landroid/view/WindowManagerPolicy$WindowState;
  
!     invoke-interface {v5, v6}, Landroid/view/WindowManagerPolicy$WindowState;->getNeedsMenuLw(Landroid/view/WindowManagerPolicy$WindowState;)Z
  
      move-result v1
  
--- 2704,2710 ----
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mTopFullscreenOpaqueWindowState:Landroid/view/WindowManagerPolicy$WindowState;
  
!     invoke-static {v5, v6}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->getNeedsMenuLw(Landroid/view/WindowManagerPolicy$WindowState;Landroid/view/WindowManagerPolicy$WindowState;)Z
  
      move-result v1
  
***************
*** 2747,2775 ****
      .parameter "screenOnListener"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_1
  
      if-eqz p1, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      new-instance v1, Lcom/android/internal/policy/impl/PhoneWindowManager$18;
  
      invoke-direct {v1, p0, p1}, Lcom/android/internal/policy/impl/PhoneWindowManager$18;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/view/WindowManagerPolicy$ScreenOnListener;)V
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onScreenTurnedOn(Lcom/android/internal/policy/impl/keyguard/KeyguardViewManager$ShowListener;)V
  
      :goto_0
      return-void
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      const/4 v1, 0x0
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onScreenTurnedOn(Lcom/android/internal/policy/impl/keyguard/KeyguardViewManager$ShowListener;)V
  
      :goto_1
      invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->waitForTopFullscreenWindowDrawn(Landroid/view/WindowManagerPolicy$ScreenOnListener;)V
--- 2758,2786 ----
      .parameter "screenOnListener"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_1
  
      if-eqz p1, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      new-instance v1, Lcom/android/internal/policy/impl/PhoneWindowManager$18;
  
      invoke-direct {v1, p0, p1}, Lcom/android/internal/policy/impl/PhoneWindowManager$18;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/view/WindowManagerPolicy$ScreenOnListener;)V
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onScreenTurnedOn(Lcom/android/internal/policy/impl/KeyguardViewManager$ShowListener;)V
  
      :goto_0
      return-void
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      const/4 v1, 0x0
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onScreenTurnedOn(Lcom/android/internal/policy/impl/KeyguardViewManager$ShowListener;)V
  
      :goto_1
      invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->waitForTopFullscreenWindowDrawn(Landroid/view/WindowManagerPolicy$ScreenOnListener;)V
***************
*** 3655,3660 ****
--- 3666,3673 ----
  
      .end local v0           #applyWindow:Z
      :cond_a
+     invoke-static {p0, p1, p2}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->setAboveStatusBarFullScreenWindow(Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/view/WindowManagerPolicy$WindowState;Landroid/view/WindowManager$LayoutParams;)V
+ 
      return-void
  
      :cond_b
***************
*** 4678,4697 ****
  
      if-nez v2, :cond_6
  
-     move-object/from16 v0, p0
- 
-     iget v2, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mUnrestrictedScreenTop:I
- 
-     move-object/from16 v0, p0
- 
-     iget v3, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarHeight:I
- 
-     add-int/2addr v2, v3
- 
-     move-object/from16 v0, p0
- 
-     iput v2, v0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mSystemTop:I
- 
      :cond_6
      const/4 v2, 0x1
  
--- 4691,4696 ----
***************
*** 4906,4911 ****
--- 4905,4922 ----
  
      iput-boolean v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mShowingDream:Z
  
+     invoke-static {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->clearAboveStatusBarFullScreenWindow(Lcom/android/internal/policy/impl/PhoneWindowManager;)V
+ 
+     return-void
+ .end method
+ 
+ .method callInterceptPowerKeyUp(Z)V
+     .locals 0
+     .parameter "canceled"
+ 
+     .prologue
+     invoke-direct {p0, p1}, Lcom/android/internal/policy/impl/PhoneWindowManager;->interceptPowerKeyUp(Z)Z
+ 
      return-void
  .end method
  
***************
*** 5132,5140 ****
      .locals 2
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowing()Z
  
      move-result v0
  
--- 5143,5151 ----
      .locals 2
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowing()Z
  
      move-result v0
  
***************
*** 7769,7781 ****
      .parameter "enabled"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->setKeyguardEnabled(Z)V
  
      :cond_0
      return-void
--- 7780,7792 ----
      .parameter "enabled"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->setKeyguardEnabled(Z)V
  
      :cond_0
      return-void
***************
*** 7801,7813 ****
      .parameter "callback"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->verifyUnlock(Landroid/view/WindowManagerPolicy$OnKeyguardExitResult;)V
  
      :cond_0
      return-void
--- 7812,7824 ----
      .parameter "callback"
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->verifyUnlock(Landroid/view/WindowManagerPolicy$OnKeyguardExitResult;)V
  
      :cond_0
      return-void
***************
*** 7914,7919 ****
--- 7925,7932 ----
      invoke-static {v6, v7}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
  
      :cond_1
+     goto :goto_miui_0
+     
      invoke-static {}, Lcom/android/internal/policy/impl/keyguard/KeyguardUpdateMonitor;->isAlarmBoot()Z
  
      move-result v6
***************
*** 7943,7948 ****
--- 7956,7968 ----
  
      :cond_3
      :goto_1
+     iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBar:Landroid/view/WindowManagerPolicy$WindowState;
+ 
+     if-eqz v6, :cond_miui_2
+     
+     invoke-static {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->checkStatusBarVisibility(Lcom/android/internal/policy/impl/PhoneWindowManager;)V
+ 
+     :cond_miui_2
      iput-boolean v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mTopIsFullscreen:Z
  
      iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguard:Landroid/view/WindowManagerPolicy$WindowState;
***************
*** 7989,7997 ****
  
      move-result-object v7
  
!     iget-object v8, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v8}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isSecure()Z
  
      move-result v8
  
--- 8009,8017 ----
  
      move-result-object v7
  
!     iget-object v8, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v8}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isSecure()Z
  
      move-result v8
  
***************
*** 8010,8018 ****
  
      if-eqz v6, :cond_14
  
!     iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v6}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isSecure()Z
  
      move-result v6
  
--- 8030,8038 ----
  
      if-eqz v6, :cond_14
  
!     iget-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v6}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isSecure()Z
  
      move-result v6
  
***************
*** 8029,8037 ****
      or-int/lit8 v0, v0, 0x7
  
      :cond_5
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v3}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowing()Z
  
      move-result v3
  
--- 8049,8057 ----
      or-int/lit8 v0, v0, 0x7
  
      :cond_5
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v3}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowing()Z
  
      move-result v3
  
***************
*** 8069,8074 ****
--- 8089,8095 ----
      goto/16 :goto_0
  
      .restart local v1       #lp:Landroid/view/WindowManager$LayoutParams;
+     :goto_miui_0
      :cond_9
      iget-boolean v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mForceStatusBar:Z
  
***************
*** 8096,8105 ****
  
      move-result v6
  
!     if-eqz v6, :cond_3
  
      or-int/lit8 v0, v0, 0x1
  
      goto/16 :goto_1
  
      :cond_c
--- 8117,8129 ----
  
      move-result v6
  
!     if-eqz v6, :cond_miui_1
  
      or-int/lit8 v0, v0, 0x1
  
+     :cond_miui_1
+     invoke-virtual {p0, v5}, Lcom/android/internal/policy/impl/PhoneWindowManager;->notifyStatusBarShowingOrHiding(Z)V
+     
      goto/16 :goto_1
  
      :cond_c
***************
*** 8247,8252 ****
--- 8271,8278 ----
  
      invoke-virtual {v6, v7}, Landroid/os/Handler;->post(Ljava/lang/Runnable;)Z
  
+     invoke-virtual {p0, v4}, Lcom/android/internal/policy/impl/PhoneWindowManager;->notifyStatusBarShowingOrHiding(Z)V
+ 
      goto/16 :goto_1
  
      :cond_10
***************
*** 8285,8294 ****
  
      move-result v6
  
!     if-eqz v6, :cond_3
  
      or-int/lit8 v0, v0, 0x1
  
      goto/16 :goto_1
  
      :cond_14
--- 8311,8323 ----
  
      move-result v6
  
!     if-eqz v6, :cond_miui_0
  
      or-int/lit8 v0, v0, 0x1
  
+     :cond_miui_0
+     invoke-virtual {p0, v5}, Lcom/android/internal/policy/impl/PhoneWindowManager;->notifyStatusBarShowingOrHiding(Z)V
+ 
      goto/16 :goto_1
  
      :cond_14
***************
*** 8307,8315 ****
      or-int/lit8 v0, v0, 0x7
  
      :cond_15
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v3, v5}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->setHidden(Z)V
  
      goto/16 :goto_2
  
--- 8336,8344 ----
      or-int/lit8 v0, v0, 0x7
  
      :cond_15
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v3, v5}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->setHidden(Z)V
  
      goto/16 :goto_2
  
***************
*** 8333,8341 ****
      or-int/lit8 v0, v0, 0x7
  
      :cond_17
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->setHidden(Z)V
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mHandler:Landroid/os/Handler;
  
--- 8362,8370 ----
      or-int/lit8 v0, v0, 0x7
  
      :cond_17
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->setHidden(Z)V
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mHandler:Landroid/os/Handler;
  
***************
*** 8361,8369 ****
      or-int/lit8 v0, v0, 0x7
  
      :cond_19
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->setHidden(Z)V
  
      goto/16 :goto_2
  .end method
--- 8390,8398 ----
      or-int/lit8 v0, v0, 0x7
  
      :cond_19
!     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->setHidden(Z)V
  
      goto/16 :goto_2
  .end method
***************
*** 8622,8627 ****
--- 8651,8665 ----
      goto :goto_1
  .end method
  
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getMaxWallpaperLayer()I
      .locals 1
  
***************
*** 8707,8712 ****
--- 8745,8768 ----
      return p1
  .end method
  
+ .method getPowerLongPress()Ljava/lang/Runnable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPowerLongPress:Ljava/lang/Runnable;
+ 
+     return-object v0
+ .end method
+ 
+ .method getScreenshotChordLongPress()Ljava/lang/Runnable;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mScreenshotChordLongPress:Ljava/lang/Runnable;
+ 
+     return-object v0
+ .end method
+ 
  .method getStatusBarService()Lcom/android/internal/statusbar/IStatusBarService;
      .locals 2
  
***************
*** 9029,9035 ****
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-nez v0, :cond_0
  
--- 9085,9091 ----
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-nez v0, :cond_0
  
***************
*** 9039,9047 ****
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isInputRestricted()Z
  
      move-result v0
  
--- 9095,9103 ----
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isInputRestricted()Z
  
      move-result v0
  
***************
*** 9081,9093 ****
  
      if-nez v6, :cond_0
  
!     new-instance v6, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
! 
!     const/4 v7, 0x0
  
!     invoke-direct {v6, p1, v7}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;-><init>(Landroid/content/Context;Lcom/android/internal/widget/LockPatternUtils;)V
  
!     iput-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      :cond_0
      new-instance v6, Lcom/android/internal/policy/impl/PhoneWindowManager$PolicyHandler;
--- 9137,9147 ----
  
      if-nez v6, :cond_0
  
!     new-instance v6, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-direct {v6, p1, p0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;-><init>(Landroid/content/Context;Lcom/android/internal/policy/impl/PhoneWindowManager;)V
  
!     iput-object v6, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      :cond_0
      new-instance v6, Lcom/android/internal/policy/impl/PhoneWindowManager$PolicyHandler;
***************
*** 11229,11255 ****
      return v10
  
      :cond_1
-     const/16 v12, 0x1a
- 
-     invoke-virtual/range {p1 .. p1}, Landroid/view/KeyEvent;->getKeyCode()I
- 
-     move-result v13
- 
-     if-ne v12, v13, :cond_2
- 
-     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
- 
-     invoke-virtual {v12}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isAlarmUnlockScreen()Z
- 
-     move-result v12
- 
-     if-eqz v12, :cond_2
- 
-     const/4 v10, 0x0
- 
-     goto :goto_0
- 
-     :cond_2
      iget-object v13, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyDispatchLock:Ljava/lang/Object;
  
      monitor-enter v13
--- 11283,11288 ----
***************
*** 11312,11318 ****
  
      .local v5, isInjected:Z
      :goto_2
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-nez v12, :cond_b
  
--- 11345,11351 ----
  
      .local v5, isInjected:Z
      :goto_2
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-nez v12, :cond_b
  
***************
*** 11423,11429 ****
  
      invoke-static {v12, v13}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
!     if-eqz v2, :cond_5
  
      move/from16 v0, p2
  
--- 11456,11462 ----
  
      invoke-static {v12, v13}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
  
!     invoke-static/range {p0 .. p2}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->performReleaseHapticFeedback(Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/view/KeyEvent;I)V
  
      move/from16 v0, p2
  
***************
*** 11440,11446 ****
--- 11473,11484 ----
      const/4 v12, 0x0
  
      const/4 v13, 0x1
+     
+     if-nez v2, :cond_miui_0
  
+     const/4 v13, 0x2
+     
+     :cond_miui_0
      const/4 v14, 0x0
  
      invoke-virtual {p0, v12, v13, v14}, Lcom/android/internal/policy/impl/PhoneWindowManager;->performHapticFeedbackLw(Landroid/view/WindowManagerPolicy$WindowState;IZ)Z
***************
*** 11502,11519 ****
      :cond_b
      if-eqz p3, :cond_c
  
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v12}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v8
  
      goto/16 :goto_3
  
      :cond_c
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v12}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowing()Z
  
      move-result v8
  
--- 11540,11557 ----
      :cond_b
      if-eqz p3, :cond_c
  
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v12}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v8
  
      goto/16 :goto_3
  
      :cond_c
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v12}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowing()Z
  
      move-result v8
  
***************
*** 11542,11550 ****
  
      if-eqz v8, :cond_f
  
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v12, v7}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onWakeKeyWhenKeyguardShowingTq(I)V
  
      goto :goto_5
  
--- 11580,11588 ----
  
      if-eqz v8, :cond_f
  
!     iget-object v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v12, v7}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onWakeKeyWhenKeyguardShowingTq(I)V
  
      goto :goto_5
  
***************
*** 11899,11904 ****
--- 11937,11944 ----
  
      iput-boolean v12, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPendingPowerKeyUpCanceled:Z
  
+     invoke-static {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->sendPowerUpBroadcast(Lcom/android/internal/policy/impl/PhoneWindowManager;)V
+ 
      goto/16 :goto_0
  
      :cond_20
***************
*** 12076,12096 ****
      :goto_0
      if-eqz v0, :cond_0
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v2, :cond_2
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowing()Z
  
      move-result v2
  
      if-eqz v2, :cond_2
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onWakeMotionWhenKeyguardShowingTq()V
  
      :cond_0
      :goto_1
--- 12116,12136 ----
      :goto_0
      if-eqz v0, :cond_0
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v2, :cond_2
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowing()Z
  
      move-result v2
  
      if-eqz v2, :cond_2
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onWakeMotionWhenKeyguardShowingTq()Z
  
      :cond_0
      :goto_1
***************
*** 12186,12192 ****
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-nez v0, :cond_0
  
--- 12226,12232 ----
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-nez v0, :cond_0
  
***************
*** 12196,12204 ****
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isSecure()Z
  
      move-result v0
  
--- 12236,12244 ----
      return v0
  
      :cond_0
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isSecure()Z
  
      move-result v0
  
***************
*** 12270,12282 ****
      .locals 4
  
      .prologue
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v2, :cond_0
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v2
  
--- 12310,12322 ----
      .locals 4
  
      .prologue
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v2, :cond_0
  
!     iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v2}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v2
  
***************
*** 12330,12342 ****
      .locals 2
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v0
  
--- 12370,12382 ----
      .locals 2
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v0
  
***************
*** 12350,12370 ****
  
      if-nez v0, :cond_1
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isInputRestricted()Z
  
      move-result v0
  
      if-eqz v0, :cond_1
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      new-instance v1, Lcom/android/internal/policy/impl/PhoneWindowManager$5;
  
      invoke-direct {v1, p0}, Lcom/android/internal/policy/impl/PhoneWindowManager$5;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager;)V
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->verifyUnlock(Landroid/view/WindowManagerPolicy$OnKeyguardExitResult;)V
  
      goto :goto_0
  
--- 12390,12410 ----
  
      if-nez v0, :cond_1
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isInputRestricted()Z
  
      move-result v0
  
      if-eqz v0, :cond_1
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      new-instance v1, Lcom/android/internal/policy/impl/PhoneWindowManager$5;
  
      invoke-direct {v1, p0}, Lcom/android/internal/policy/impl/PhoneWindowManager$5;-><init>(Lcom/android/internal/policy/impl/PhoneWindowManager;)V
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->verifyUnlock(Landroid/view/WindowManagerPolicy$OnKeyguardExitResult;)V
  
      goto :goto_0
  
***************
*** 14698,14708 ****
  
      if-eqz v1, :cond_3
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      const/16 v2, 0x1a
  
!     invoke-virtual {v1, v2}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onWakeKeyWhenKeyguardShowingTq(I)V
  
      goto :goto_0
  
--- 14738,14748 ----
  
      if-eqz v1, :cond_3
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      const/16 v2, 0x1a
  
!     invoke-virtual {v1, v2}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onWakeKeyWhenKeyguardShowingTq(I)V
  
      goto :goto_0
  
***************
*** 14740,14745 ****
--- 14780,14839 ----
      goto :goto_0
  .end method
  
+ .method notifyStatusBarShowingOrHiding(Z)V
+     .locals 5
+     .parameter "showing"
+ 
+     .prologue
+     :try_start_0
+     invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->getStatusBarService()Lcom/android/internal/statusbar/IStatusBarService;
+ 
+     move-result-object v1
+ 
+     .local v1, statusbar:Lcom/android/internal/statusbar/IStatusBarService;
+     if-eqz p1, :cond_0
+ 
+     const/4 v2, 0x0
+ 
+     :goto_0
+     iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarDisableToken:Landroid/os/IBinder;
+ 
+     const-string v4, "system"
+ 
+     invoke-interface {v1, v2, v3, v4}, Lcom/android/internal/statusbar/IStatusBarService;->disable(ILandroid/os/IBinder;Ljava/lang/String;)V
+     :try_end_0
+     .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
+ 
+     .end local v1           #statusbar:Lcom/android/internal/statusbar/IStatusBarService;
+     :goto_1
+     return-void
+ 
+     .restart local v1       #statusbar:Lcom/android/internal/statusbar/IStatusBarService;
+     :cond_0
+     const/high16 v2, 0x2000
+ 
+     goto :goto_0
+ 
+     .end local v1           #statusbar:Lcom/android/internal/statusbar/IStatusBarService;
+     :catch_0
+     move-exception v0
+ 
+     .local v0, e:Landroid/os/RemoteException;
+     const/4 v2, 0x0
+ 
+     iput-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
+ 
+     goto :goto_1
+ .end method
+ 
+ .method onScreenShotMessageSend(Landroid/os/Message;)V
+     .locals 0
+     .parameter "msg"
+ 
+     .prologue
+     return-void
+ .end method
+ 
  .method public performHapticFeedbackLw(Landroid/view/WindowManagerPolicy$WindowState;IZ)Z
      .locals 7
      .parameter "win"
***************
*** 14788,14796 ****
  
      if-nez v0, :cond_0
  
!     iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v4}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v4
  
--- 14882,14890 ----
  
      if-nez v0, :cond_0
  
!     iget-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v4}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->isShowingAndNotHidden()Z
  
      move-result v4
  
***************
*** 15751,15763 ****
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onScreenTurnedOff(I)V
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
--- 15845,15857 ----
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onScreenTurnedOff(I)V
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
***************
*** 16204,16216 ****
      .prologue
      const/4 v1, 0x0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->setCurrentUser(I)V
  
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
--- 16298,16310 ----
      .prologue
      const/4 v1, 0x0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0, p1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->setCurrentUser(I)V
  
      :cond_0
      iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mStatusBarService:Lcom/android/internal/statusbar/IStatusBarService;
***************
*** 16842,16847 ****
--- 16936,16958 ----
      return-void
  .end method
  
+ .method setMaxBacklightBrightness()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method setPowerLongPress(Ljava/lang/Runnable;)V
+     .locals 0
+     .parameter "value"
+ 
+     .prologue
+     iput-object p1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPowerLongPress:Ljava/lang/Runnable;
+ 
+     return-void
+ .end method
+ 
  .method public setRotationLw(I)V
      .locals 1
      .parameter "rotation"
***************
*** 16924,16932 ****
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->showAssistant()V
  
      return-void
  .end method
--- 17035,17043 ----
      .locals 1
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->showAssistant()V
  
      return-void
  .end method
***************
*** 16960,16978 ****
      .locals 4
  
      .prologue
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      if-nez v1, :cond_0
  
!     new-instance v1, Lcom/android/internal/policy/impl/GlobalActions;
  
      iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mWindowManagerFuncs:Landroid/view/WindowManagerPolicy$WindowManagerFuncs;
  
!     invoke-direct {v1, v2, v3}, Lcom/android/internal/policy/impl/GlobalActions;-><init>(Landroid/content/Context;Landroid/view/WindowManagerPolicy$WindowManagerFuncs;)V
  
!     iput-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      :cond_0
      invoke-direct {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->keyguardIsShowingTq()Z
--- 17071,17089 ----
      .locals 4
  
      .prologue
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      if-nez v1, :cond_0
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mWindowManagerFuncs:Landroid/view/WindowManagerPolicy$WindowManagerFuncs;
  
!     invoke-direct {v1, v2, v3}, Lcom/android/internal/policy/impl/MiuiGlobalActions;-><init>(Landroid/content/Context;Landroid/view/WindowManagerPolicy$WindowManagerFuncs;)V
  
!     iput-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      :cond_0
      invoke-direct {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->keyguardIsShowingTq()Z
***************
*** 16980,16998 ****
      move-result v0
  
      .local v0, keyguardShowing:Z
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->isDeviceProvisioned()Z
  
      move-result v2
  
!     invoke-virtual {v1, v0, v2}, Lcom/android/internal/policy/impl/GlobalActions;->showDialog(ZZ)V
  
      if-eqz v0, :cond_1
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v1}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->userActivity()V
  
      :cond_1
      return-void
--- 17091,17109 ----
      move-result v0
  
      .local v0, keyguardShowing:Z
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->isDeviceProvisioned()Z
  
      move-result v2
  
!     invoke-virtual {v1, v0, v2}, Lcom/android/internal/policy/impl/MiuiGlobalActions;->showDialog(ZZ)V
  
      if-eqz v0, :cond_1
  
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v1}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->userActivity()V
  
      :cond_1
      return-void
***************
*** 17126,17138 ****
      .locals 3
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/keyguard/KeyguardViewMediator;->onSystemReady()V
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
--- 17237,17249 ----
      .locals 3
  
      .prologue
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
      if-eqz v0, :cond_0
  
!     iget-object v0, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;
  
!     invoke-virtual {v0}, Lcom/android/internal/policy/impl/MiuiKeyguardViewMediator;->onSystemReady()V
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mLock:Ljava/lang/Object;
***************
*** 17635,17641 ****
  .end method
  
  .method public windowTypeToLayerLw(I)I
!     .locals 4
      .parameter "type"
  
      .prologue
--- 17746,17752 ----
  .end method
  
  .method public windowTypeToLayerLw(I)I
!     .locals 3
      .parameter "type"
  
      .prologue
***************
*** 17656,17682 ****
      :cond_0
      packed-switch p1, :pswitch_data_0
  
!     const-string v1, "WindowManager"
! 
!     new-instance v2, Ljava/lang/StringBuilder;
! 
!     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
! 
!     const-string v3, "Unknown window type: "
  
!     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     invoke-virtual {v2, p1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
! 
!     move-result-object v2
! 
!     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
! 
!     move-result-object v2
! 
!     invoke-static {v1, v2}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
  
      goto :goto_0
  
--- 17767,17775 ----
      :cond_0
      packed-switch p1, :pswitch_data_0
  
!     invoke-static {p1, v0}, Lcom/android/internal/policy/impl/PhoneWindowManager$Injector;->getMiuiViewLayer(II)I
  
!     move-result v0
  
      goto :goto_0
  
