*** ./smali/android/media/MediaFile.smali	2014-10-11 10:20:43.716208908 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/media/MediaFile.smali	2014-10-11 10:17:40.952204944 +1100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/media/MediaFile$MediaFileType;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/media/MediaFile$MediaFileType;,
!         Landroid/media/MediaFile$Injector;
      }
  .end annotation
  
***************
*** 22,28 ****
  
  .field public static final FILE_TYPE_AMR:I = 0x68
  
! .field public static final FILE_TYPE_APE:I = 0x6f
  
  .field public static final FILE_TYPE_APK:I = 0x31f
  
--- 23,29 ----
  
  .field public static final FILE_TYPE_AMR:I = 0x68
  
! .field public static final FILE_TYPE_APE:I = 0x3e9
  
  .field public static final FILE_TYPE_APK:I = 0x31f
  
***************
*** 40,46 ****
  
  .field public static final FILE_TYPE_FLAC:I = 0x6e
  
! .field public static final FILE_TYPE_FLV:I = 0x18e
  
  .field public static final FILE_TYPE_GIF:I = 0x192
  
--- 41,47 ----
  
  .field public static final FILE_TYPE_FLAC:I = 0x6e
  
! .field public static final FILE_TYPE_FLV:I = 0xca
  
  .field public static final FILE_TYPE_GIF:I = 0x192
  
***************
*** 138,143 ****
--- 139,146 ----
  
  .field private static final FIRST_DRM_FILE_TYPE:I = 0x259
  
+ .field private static final FIRST_FFMPEG_AUDIO_FILE_TYPE:I = 0x3e9
+ 
  .field private static final FIRST_IMAGE_FILE_TYPE:I = 0x191
  
  .field private static final FIRST_MIDI_FILE_TYPE:I = 0xc9
***************
*** 154,159 ****
--- 157,164 ----
  
  .field private static final LAST_DRM_FILE_TYPE:I = 0x259
  
+ .field private static final LAST_FFMPEG_AUDIO_FILE_TYPE:I = 0x3e9
+ 
  .field private static final LAST_IMAGE_FILE_TYPE:I = 0x196
  
  .field private static final LAST_MIDI_FILE_TYPE:I = 0xcb
***************
*** 164,170 ****
  
  .field private static final LAST_VIDEO_FILE_TYPE:I = 0x18f
  
! .field private static final LAST_VIDEO_FILE_TYPE2:I = 0x320
  
  .field private static final sFileTypeMap:Ljava/util/HashMap;
      .annotation system Ldalvik/annotation/Signature;
--- 169,175 ----
  
  .field private static final LAST_VIDEO_FILE_TYPE:I = 0x18f
  
! .field private static final LAST_VIDEO_FILE_TYPE2:I = 0xca
  
  .field private static final sFileTypeMap:Ljava/util/HashMap;
      .annotation system Ldalvik/annotation/Signature;
***************
*** 294,300 ****
  
      const-string v0, "APE"
  
!     const/16 v1, 0x6f
  
      const-string v2, "audio/ape"
  
--- 299,305 ----
  
      const-string v0, "APE"
  
!     const/16 v1, 0x3e9
  
      const-string v2, "audio/ape"
  
***************
*** 374,380 ****
  
      invoke-static {v0, v1, v2}, Landroid/media/MediaFile;->addFileType(Ljava/lang/String;ILjava/lang/String;)V
  
!     invoke-static {}, Landroid/media/MediaFile;->isWMAEnabled()Z
  
      move-result v0
  
--- 379,385 ----
  
      invoke-static {v0, v1, v2}, Landroid/media/MediaFile;->addFileType(Ljava/lang/String;ILjava/lang/String;)V
  
!     invoke-static {}, Landroid/media/MediaFile$Injector;->isWMAEnabled()Z
  
      move-result v0
  
***************
*** 503,508 ****
--- 508,515 ----
  
      invoke-static {v0, v4, v1}, Landroid/media/MediaFile;->addFileType(Ljava/lang/String;ILjava/lang/String;)V
  
+     invoke-static {}, Landroid/media/MediaFile$Injector;->addAPE()V
+ 
      const-string v0, "MPEG"
  
      const/16 v1, 0x12d
***************
*** 633,639 ****
  
      const-string v0, "FLV"
  
!     const/16 v1, 0x18e
  
      const-string v2, "video/x-flv"
  
--- 640,646 ----
  
      const-string v0, "FLV"
  
!     const/16 v1, 0xca
  
      const-string v2, "video/x-flv"
  
***************
*** 1006,1011 ****
--- 1013,1020 ----
  
      invoke-static {v0, v8, v1}, Landroid/media/MediaFile;->addFileType(Ljava/lang/String;ILjava/lang/String;)V
  
+     invoke-static {}, Landroid/media/MediaFile$Injector;->addFLV()V
+ 
      return-void
  .end method
  
***************
*** 1360,1393 ****
  .end method
  
  .method public static isAudioFileType(I)Z
!     .locals 1
      .parameter "fileType"
  
      .prologue
      const/16 v0, 0x65
  
      if-lt p0, v0, :cond_0
  
      const/16 v0, 0xc7
  
!     if-le p0, v0, :cond_1
  
      :cond_0
      const/16 v0, 0xc9
  
!     if-lt p0, v0, :cond_2
  
      const/16 v0, 0xcb
  
!     if-gt p0, v0, :cond_2
  
      :cond_1
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_2
      const/4 v0, 0x0
  
      goto :goto_0
--- 1369,1409 ----
  .end method
  
  .method public static isAudioFileType(I)Z
!     .locals 2
      .parameter "fileType"
  
      .prologue
+     const/16 v1, 0x3e9
+ 
      const/16 v0, 0x65
  
      if-lt p0, v0, :cond_0
  
      const/16 v0, 0xc7
  
!     if-le p0, v0, :cond_2
  
      :cond_0
      const/16 v0, 0xc9
  
!     if-lt p0, v0, :cond_1
  
      const/16 v0, 0xcb
  
!     if-le p0, v0, :cond_2
  
      :cond_1
+     if-lt p0, v1, :cond_3
+ 
+     if-gt p0, v1, :cond_3
+ 
+     :cond_2
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_3
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 1520,1531 ****
  .end method
  
  .method public static isVideoFileType(I)Z
!     .locals 2
      .parameter "fileType"
  
      .prologue
-     const/16 v1, 0x320
- 
      const/16 v0, 0x12d
  
      if-lt p0, v0, :cond_0
--- 1536,1545 ----
  .end method
  
  .method public static isVideoFileType(I)Z
!     .locals 1
      .parameter "fileType"
  
      .prologue
      const/16 v0, 0x12d
  
      if-lt p0, v0, :cond_0
***************
*** 1535,1543 ****
      if-le p0, v0, :cond_1
  
      :cond_0
!     if-lt p0, v1, :cond_2
  
!     if-gt p0, v1, :cond_2
  
      :cond_1
      const/4 v0, 0x1
--- 1549,1561 ----
      if-le p0, v0, :cond_1
  
      :cond_0
!     const/16 v0, 0x320
! 
!     if-lt p0, v0, :cond_2
  
!     const/16 v0, 0xca
! 
!     if-gt p0, v0, :cond_2
  
      :cond_1
      const/4 v0, 0x1
