*** ./smali/com/android/internal/widget/LockPatternUtils.smali	2014-10-11 10:20:06.812208108 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/LockPatternUtils.smali	2014-10-11 10:17:05.976204185 +1100
***************
*** 34,39 ****
--- 34,41 ----
  
  .field protected static final LOCKOUT_PERMANENT_KEY:Ljava/lang/String; = "lockscreen.lockedoutpermanently"
  
+ .field public static final LOCKSCREEN_BIOMETRIC_WEAK_FALLBACK:Ljava/lang/String; = "lockscreen.biometric_weak_fallback"
+ 
  .field protected static final LOCKSCREEN_OPTIONS:Ljava/lang/String; = "lockscreen.options"
  
  .field public static final LOCKSCREEN_POWER_BUTTON_INSTANTLY_LOCKS:Ljava/lang/String; = "lockscreen.power_button_instantly_locks"
***************
*** 133,138 ****
--- 135,162 ----
      return-void
  .end method
  
+ .method public static callPatternToHash(Ljava/util/List;)[B
+     .locals 1
+     .parameter
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "(",
+             "Ljava/util/List",
+             "<",
+             "Lcom/android/internal/widget/LockPatternView$Cell;",
+             ">;)[B"
+         }
+     .end annotation
+ 
+     .prologue
+     .local p0, pattern:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/widget/LockPatternView$Cell;>;"
+     invoke-static {p0}, Lcom/android/internal/widget/LockPatternUtils;->patternToHash(Ljava/util/List;)[B
+ 
+     move-result-object v0
+ 
+     return-object v0
+ .end method
+ 
  .method private static combineStrings([ILjava/lang/String;)Ljava/lang/String;
      .locals 7
      .parameter "list"
***************
*** 474,480 ****
      return-object v0
  .end method
  
! .method private getLong(Ljava/lang/String;J)J
      .locals 3
      .parameter "secureSettingKey"
      .parameter "defaultValue"
--- 498,504 ----
      return-object v0
  .end method
  
! .method protected getLong(Ljava/lang/String;J)J
      .locals 3
      .parameter "secureSettingKey"
      .parameter "defaultValue"
***************
*** 515,521 ****
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-direct {p0, v3, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 539,545 ----
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-virtual {p0, v3, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 537,543 ****
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-direct {p0, v3, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "LockPatternUtils"
  
--- 561,567 ----
  
      const-string v3, "lockscreen.password_salt"
  
!     invoke-virtual {p0, v3, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v3, "LockPatternUtils"
  
***************
*** 869,875 ****
      goto :goto_0
  .end method
  
! .method private setLong(Ljava/lang/String;J)V
      .locals 1
      .parameter "secureSettingKey"
      .parameter "value"
--- 893,899 ----
      goto :goto_0
  .end method
  
! .method protected setLong(Ljava/lang/String;J)V
      .locals 1
      .parameter "secureSettingKey"
      .parameter "value"
***************
*** 1505,1515 ****
  
      const-string v0, "lockscreen.password_type"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v0, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  .end method
--- 1529,1539 ----
  
      const-string v0, "lockscreen.password_type"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v0, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  .end method
***************
*** 1642,1648 ****
  
      const-wide/32 v3, 0x10000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v2
  
--- 1666,1672 ----
  
      const-wide/32 v3, 0x10000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v2
  
***************
*** 1956,1961 ****
--- 1980,1995 ----
      return v0
  .end method
  
+ .method public getKeyguardLockoutAttemptDeadline(I)J
+     .locals 2
+     .parameter "failedAttempts"
+ 
+     .prologue
+     const-wide/16 v0, -0x1
+ 
+     return-wide v0
+ .end method
+ 
  .method public getKeyguardStoredPasswordQuality()I
      .locals 5
  
***************
*** 1964,1970 ****
  
      const-string v1, "lockscreen.password_type"
  
!     invoke-direct {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 1998,2004 ----
  
      const-string v1, "lockscreen.password_type"
  
!     invoke-virtual {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 1982,1988 ****
      :cond_0
      const-string v1, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 2016,2022 ----
      :cond_0
      const-string v1, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v1, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 2000,2006 ****
  
      const-string v6, "lockscreen.lockoutattemptdeadline"
  
!     invoke-direct {p0, v6, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 2034,2040 ----
  
      const-string v6, "lockscreen.lockoutattemptdeadline"
  
!     invoke-virtual {p0, v6, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 2368,2374 ****
  
      const-string v2, "lock_biometric_weak_flags"
  
!     invoke-direct {p0, v2, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 2402,2408 ----
  
      const-string v2, "lock_biometric_weak_flags"
  
!     invoke-virtual {p0, v2, v4, v5}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 2446,2459 ****
  
      const-string v8, "lockscreen.password_type"
  
!     invoke-direct {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
      .local v3, mode:J
      const-string v8, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 2480,2493 ----
  
      const-string v8, "lockscreen.password_type"
  
!     invoke-virtual {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
      .local v3, mode:J
      const-string v8, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v8, v9, v10}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 2567,2573 ****
  
      const-string v3, "lockscreen.password_type_alternate"
  
!     invoke-direct {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
--- 2601,2607 ----
  
      const-string v3, "lockscreen.password_type_alternate"
  
!     invoke-virtual {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
***************
*** 2589,2595 ****
  
      const-string v3, "lockscreen.password_type"
  
!     invoke-direct {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
--- 2623,2629 ----
  
      const-string v3, "lockscreen.password_type"
  
!     invoke-virtual {p0, v3, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v3
  
***************
*** 2635,2640 ****
--- 2669,2678 ----
      .prologue
      const-wide/16 v2, 0x0
  
+     sget-boolean v0, Lmiui/os/Build;->IS_ALPHA_BUILD:Z
+ 
+     if-nez v0, :cond_0
+ 
      invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils;->isSecure()Z
  
      move-result v0
***************
*** 2643,2649 ****
  
      const-string v0, "lockscreen.disabled"
  
!     invoke-direct {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 2681,2687 ----
  
      const-string v0, "lockscreen.disabled"
  
!     invoke-virtual {p0, v0, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 3120,3128 ****
  .end method
  
  .method public reportSuccessfulPasswordAttempt()V
!     .locals 2
  
      .prologue
      invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils;->getDevicePolicyManager()Landroid/app/admin/DevicePolicyManager;
  
      move-result-object v0
--- 3158,3172 ----
  .end method
  
  .method public reportSuccessfulPasswordAttempt()V
!     .locals 3
  
      .prologue
+     const-string v0, "lockscreen.lockoutattemptdeadline"
+ 
+     const-wide/16 v1, 0x0
+ 
+     invoke-virtual {p0, v0, v1, v2}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
+ 
      invoke-virtual {p0}, Lcom/android/internal/widget/LockPatternUtils;->getDevicePolicyManager()Landroid/app/admin/DevicePolicyManager;
  
      move-result-object v0
***************
*** 3201,3206 ****
--- 3245,3276 ----
      return-void
  .end method
  
+ .method public saveLockPassword(Ljava/lang/String;IZ)V
+     .locals 6
+     .parameter "password"
+     .parameter "quality"
+     .parameter "isFallback"
+ 
+     .prologue
+     const-string v4, "face_unlock"
+ 
+     invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->getCurrentOrCallingUserId()I
+ 
+     move-result v5
+ 
+     move-object v0, p0
+ 
+     move-object v1, p1
+ 
+     move v2, p2
+ 
+     move v3, p3
+ 
+     invoke-virtual/range {v0 .. v5}, Lcom/android/internal/widget/LockPatternUtils;->saveLockPassword(Ljava/lang/String;IZLjava/lang/String;I)V
+ 
+     return-void
+ .end method
+ 
  .method public saveLockPassword(Ljava/lang/String;IZLjava/lang/String;)V
      .locals 6
      .parameter "password"
***************
*** 3810,3815 ****
--- 3880,3910 ----
      return-void
  .end method
  
+ .method public saveLockPattern(Ljava/util/List;Z)V
+     .locals 2
+     .parameter
+     .parameter "isFallback"
+     .annotation system Ldalvik/annotation/Signature;
+         value = {
+             "(",
+             "Ljava/util/List",
+             "<",
+             "Lcom/android/internal/widget/LockPatternView$Cell;",
+             ">;Z)V"
+         }
+     .end annotation
+ 
+     .prologue
+     .local p1, pattern:Ljava/util/List;,"Ljava/util/List<Lcom/android/internal/widget/LockPatternView$Cell;>;"
+     const/4 v0, 0x0
+ 
+     const-string v1, "face_unlock"
+ 
+     invoke-virtual {p0, p1, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->saveLockPattern(Ljava/util/List;ZLjava/lang/String;)V
+ 
+     return-void
+ .end method
+ 
  .method public saveLockPattern(Ljava/util/List;ZLjava/lang/String;)V
      .locals 14
      .parameter
***************
*** 3879,3885 ****
  
      const-wide/32 v3, 0x10000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const/high16 v2, 0x1
  
--- 3974,3980 ----
  
      const-wide/32 v3, 0x10000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const/high16 v2, 0x1
  
***************
*** 3928,3940 ****
  
      const-wide/32 v3, 0x8000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v2, "lockscreen.password_type_alternate"
  
      const-wide/32 v3, 0x10000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
--- 4023,4035 ----
  
      const-wide/32 v3, 0x8000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v2, "lockscreen.password_type_alternate"
  
      const-wide/32 v3, 0x10000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishBiometricWeak()V
  
***************
*** 4013,4025 ****
  
      const-wide/16 v3, 0x4000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v2, "lockscreen.password_type_alternate"
  
      const-wide/32 v3, 0x10000
  
!     invoke-direct {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishVoiceWeak()V
  
--- 4108,4120 ----
  
      const-wide/16 v3, 0x4000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      const-string v2, "lockscreen.password_type_alternate"
  
      const-wide/32 v3, 0x10000
  
!     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      invoke-direct {p0}, Lcom/android/internal/widget/LockPatternUtils;->finishVoiceWeak()V
  
***************
*** 4155,4161 ****
  
      const-wide/16 v5, 0x0
  
!     invoke-direct {p0, v4, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
--- 4250,4256 ----
  
      const-wide/16 v5, 0x0
  
!     invoke-virtual {p0, v4, v5, v6}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v0
  
***************
*** 4170,4176 ****
      :goto_0
      const-string v4, "lock_biometric_weak_flags"
  
!     invoke-direct {p0, v4, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
--- 4265,4271 ----
      :goto_0
      const-string v4, "lock_biometric_weak_flags"
  
!     invoke-virtual {p0, v4, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
***************
*** 4194,4199 ****
--- 4289,4304 ----
      return-void
  .end method
  
+ .method public setKeyguardLockoutAttemptDeadline(I)J
+     .locals 2
+     .parameter "failedAttempts"
+ 
+     .prologue
+     const-wide/16 v0, -0x1
+ 
+     return-wide v0
+ .end method
+ 
  .method public setLockPatternEnabled(Z)V
      .locals 1
      .parameter "enabled"
***************
*** 4218,4224 ****
      const-wide/16 v0, 0x1
  
      :goto_0
!     invoke-direct {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
--- 4323,4329 ----
      const-wide/16 v0, 0x1
  
      :goto_0
!     invoke-virtual {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-void
  
***************
*** 4243,4249 ****
      .local v0, deadline:J
      const-string v2, "lockscreen.lockoutattemptdeadline"
  
!     invoke-direct {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-wide v0
  .end method
--- 4348,4354 ----
      .local v0, deadline:J
      const-string v2, "lockscreen.lockoutattemptdeadline"
  
!     invoke-virtual {p0, v2, v0, v1}, Lcom/android/internal/widget/LockPatternUtils;->setLong(Ljava/lang/String;J)V
  
      return-wide v0
  .end method
***************
*** 4272,4277 ****
--- 4377,4390 ----
      return-void
  .end method
  
+ .method public setTactileFeedbackEnabled(Z)V
+     .locals 0
+     .parameter "enabled"
+ 
+     .prologue
+     return-void
+ .end method
+ 
  .method public setVisiblePatternEnabled(Z)V
      .locals 1
      .parameter "enabled"
***************
*** 4446,4452 ****
  
      const-wide/32 v2, 0x10000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 4559,4565 ----
  
      const-wide/32 v2, 0x10000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
***************
*** 4476,4482 ****
  
      const-wide/32 v2, 0x10000
  
!     invoke-direct {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
--- 4589,4595 ----
  
      const-wide/32 v2, 0x10000
  
!     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/widget/LockPatternUtils;->getLong(Ljava/lang/String;J)J
  
      move-result-wide v1
  
