*** ./smali/com/android/internal/widget/LockSettingsService.smali	2014-10-11 10:20:06.208208095 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/LockSettingsService.smali	2014-10-11 10:17:05.756204181 +1100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/widget/LockSettingsService$DatabaseHelper;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/widget/LockSettingsService$DatabaseHelper;,
!         Lcom/android/internal/widget/LockSettingsService$Injector;
      }
  .end annotation
  
***************
*** 916,922 ****
      .restart local v3       #raf:Ljava/io/RandomAccessFile;
      .restart local v4       #stored:[B
      :cond_0
!     invoke-static {v4, p1}, Ljava/util/Arrays;->equals([B[B)Z
      :try_end_0
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
--- 917,927 ----
      .restart local v3       #raf:Ljava/io/RandomAccessFile;
      .restart local v4       #stored:[B
      :cond_0
!     invoke-static {v4, p1}, Lcom/android/internal/widget/LockSettingsService$Injector;->passwordToHash([B[B)[B
! 
!     move-result-object v6
! 
!     invoke-static {v4, v6}, Ljava/util/Arrays;->equals([B[B)Z
      :try_end_0
      .catch Ljava/io/FileNotFoundException; {:try_start_0 .. :try_end_0} :catch_0
      .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
