*** ./smali/com/android/internal/statusbar/IStatusBarService.smali	2014-10-11 10:20:09.912208175 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/statusbar/IStatusBarService.smali	2014-10-11 10:17:07.888204227 +1100
***************
*** 187,192 ****
--- 187,200 ----
      .end annotation
  .end method
  
+ .method public abstract setStatus(ILandroid/os/IBinder;Ljava/lang/String;Landroid/os/Bundle;)V
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
  .method public abstract setSystemUiVisibility(II)V
      .annotation system Ldalvik/annotation/Throws;
          value = {
