*** ./smali/android/content/ContentResolver.smali	2014-10-11 10:20:31.596208645 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/content/ContentResolver.smali	2014-10-11 10:17:27.408204650 +1100
***************
*** 2227,2232 ****
--- 2227,2241 ----
      throw v0
  .end method
  
+ .method public getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/content/ContentResolver;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getResourceId(Landroid/net/Uri;)Landroid/content/ContentResolver$OpenResourceIdResult;
      .locals 11
      .parameter "uri"
***************
*** 3865,3871 ****
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 24
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
--- 3874,3880 ----
  .end method
  
  .method public final query(Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;Landroid/os/CancellationSignal;)Landroid/database/Cursor;
!     .locals 31
      .parameter "uri"
      .parameter "projection"
      .parameter "selection"
***************
*** 3913,3918 ****
--- 3922,3928 ----
      invoke-virtual {v0, v9}, Landroid/os/CancellationSignal;->setRemote(Landroid/os/ICancellationSignal;)V
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_1
  
      :cond_2
***************
*** 3931,3936 ****
--- 3941,3947 ----
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
      .catch Landroid/os/DeadObjectException; {:try_start_1 .. :try_end_1} :catch_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_1 .. :try_end_1} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_1 .. :try_end_1} :catch_1
  
      move-result-object v19
***************
*** 3969,3974 ****
--- 3980,3986 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_2
      .catchall {:try_start_2 .. :try_end_2} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_2 .. :try_end_2} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_2 .. :try_end_2} :catch_1
  
      move-result-object v10
***************
*** 4078,4083 ****
--- 4090,4096 ----
      invoke-virtual {v0, v4, v1, v5}, Landroid/content/ContentResolver;->AddToQueryHistory(Ljava/lang/String;Ljava/lang/Throwable;I)Z
      :try_end_3
      .catchall {:try_start_3 .. :try_end_3} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_3 .. :try_end_3} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_3 .. :try_end_3} :catch_1
  
      if-eqz v3, :cond_7
***************
*** 4102,4107 ****
--- 4115,4121 ----
      invoke-virtual/range {p0 .. p1}, Landroid/content/ContentResolver;->acquireProvider(Landroid/net/Uri;)Landroid/content/IContentProvider;
      :try_end_4
      .catchall {:try_start_4 .. :try_end_4} :catchall_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_4 .. :try_end_4} :catch_miui_0
      .catch Landroid/os/RemoteException; {:try_start_4 .. :try_end_4} :catch_1
  
      move-result-object v4
***************
*** 4134,4139 ****
--- 4148,4196 ----
      goto/16 :goto_0
  
      .end local v18           #e:Landroid/os/RemoteException;
+     :catch_miui_0
+     move-exception v25
+ 
+     .local v25, e:Ljava/lang/IllegalStateException;
+     :try_start_miui_0
+     move-object/from16 v0, p0
+ 
+     iget-object v0, v0, Landroid/content/ContentResolver;->mContext:Landroid/content/Context;
+ 
+     move-object/from16 v24, v0
+ 
+     move-object/from16 v26, p1
+ 
+     move-object/from16 v27, p2
+ 
+     move-object/from16 v28, p3
+ 
+     move-object/from16 v29, p4
+ 
+     move-object/from16 v30, p5
+ 
+     invoke-static/range {v24 .. v30}, Landroid/content/ContentResolver$Injector;->checkCursorLeakException(Landroid/content/Context;Ljava/lang/IllegalStateException;Landroid/net/Uri;[Ljava/lang/String;Ljava/lang/String;[Ljava/lang/String;Ljava/lang/String;)Landroid/database/Cursor;
+     :try_end_miui_0
+     .catchall {:try_start_miui_0 .. :try_end_miui_0} :catchall_0
+ 
+     move-result-object v23
+ 
+     if-eqz v3, :cond_miui_0
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-virtual {v0, v3}, Landroid/content/ContentResolver;->releaseUnstableProvider(Landroid/content/IContentProvider;)Z
+ 
+     :cond_miui_0
+     if-eqz v10, :cond_0
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-virtual {v0, v10}, Landroid/content/ContentResolver;->releaseProvider(Landroid/content/IContentProvider;)Z
+ 
+     goto/16 :goto_0
+ 
+     .end local v25           #e:Ljava/lang/IllegalStateException;
      :catchall_0
      move-exception v4
  
