*** ./smali/android/text/SpanSet.smali	2014-10-11 10:20:49.532209034 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/text/SpanSet.smali	2014-10-11 10:17:47.292205081 +1100
***************
*** 217,223 ****
--- 217,228 ----
  
      iput-object v7, p0, Landroid/text/SpanSet;->spanFlags:[I
  
+     goto :goto_miui_0
+     
      :cond_1
+     invoke-virtual {p0}, Landroid/text/SpanSet;->recycle()V
+     
+     :goto_miui_0
      const/4 v7, 0x0
  
      iput v7, p0, Landroid/text/SpanSet;->numberOfSpans:I
