*** ./smali/com/android/server/wm/WindowManagerService.smali	2014-10-11 10:21:21.912209737 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/services.jar.out/./smali/com/android/server/wm/WindowManagerService.smali	2014-10-11 10:18:20.868205810 +1100
***************
*** 20,26 ****
          Lcom/android/server/wm/WindowManagerService$DragInputEventReceiver;,
          Lcom/android/server/wm/WindowManagerService$LayoutToAnimatorParams;,
          Lcom/android/server/wm/WindowManagerService$AppWindowAnimParams;,
!         Lcom/android/server/wm/WindowManagerService$LayoutFields;
      }
  .end annotation
  
--- 20,27 ----
          Lcom/android/server/wm/WindowManagerService$DragInputEventReceiver;,
          Lcom/android/server/wm/WindowManagerService$LayoutToAnimatorParams;,
          Lcom/android/server/wm/WindowManagerService$AppWindowAnimParams;,
!         Lcom/android/server/wm/WindowManagerService$LayoutFields;,
!         Lcom/android/server/wm/WindowManagerService$Injector;
      }
  .end annotation
  
***************
*** 9532,9537 ****
--- 9533,9547 ----
      .end local v5           #pos:I
      .end local v7           #windows:Lcom/android/server/wm/WindowList;
      :cond_6
+     invoke-static {p0, p2}, Lcom/android/server/wm/WindowManagerService$Injector;->updateFocusAndLayout(Lcom/android/server/wm/WindowManagerService;I)Z
+ 
+     move-result v8
+ 
+     if-nez v8, :cond_miui_0
+ 
+     return-void
+ 
+     :cond_miui_0
      iget-object v8, p0, Lcom/android/server/wm/WindowManagerService;->mInputMonitor:Lcom/android/server/wm/InputMonitor;
  
      invoke-virtual {v8}, Lcom/android/server/wm/InputMonitor;->setUpdateInputWindowsNeededLw()V
***************
*** 13428,13433 ****
--- 13438,13445 ----
  
      :cond_5
      :goto_5
+     invoke-static {p0}, Lcom/android/server/wm/WindowManagerService$Injector;->hideFloatingWindow(Lcom/android/server/wm/WindowManagerService;)V
+ 
      invoke-static {v9, v10}, Landroid/os/Trace;->traceEnd(J)V
  
      goto/16 :goto_0
***************
*** 26439,26444 ****
--- 26451,26467 ----
      :cond_2
      iput-boolean v5, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
  
+     const/4 v5, 0x0
+     
+     if-lez v1, :cond_miui_0
+ 
+     if-lez v4, :cond_miui_0
+     
+     const/4 v5, 0x1
+ 
+     :cond_miui_0
+     iput-boolean v5, p0, Lcom/android/server/wm/WindowManagerService;->mSafeMode:Z
+ 
      :try_start_0
      const-string v5, "persist.sys.safemode"
  
***************
*** 40003,40008 ****
--- 40026,40042 ----
      return-void
  .end method
  
+ .method public rebootSafeMode()V
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, v0}, Lcom/android/server/wm/WindowManagerService;->rebootSafeMode(Z)V
+ 
+     return-void
+ .end method
+ 
  .method public rebootSafeMode(Z)V
      .locals 1
      .parameter "confirm"
***************
*** 40792,40797 ****
--- 40826,40833 ----
      return v31
  
      :cond_1
+     invoke-static/range {v29 .. v29}, Lcom/android/server/wm/WindowManagerService$Injector;->restoreFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      move-object/from16 v0, v29
  
      iget-object v0, v0, Lcom/android/server/wm/WindowState;->mWinAnimator:Lcom/android/server/wm/WindowStateAnimator;
***************
*** 42484,42489 ****
--- 42520,42527 ----
  
      invoke-virtual {v0, v1}, Lcom/android/server/wm/InputMonitor;->updateInputWindowsLw(Z)V
  
+     invoke-static/range {v29 .. v29}, Lcom/android/server/wm/WindowManagerService$Injector;->saveFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      sget-boolean v31, Lcom/android/server/wm/WindowManagerService;->DEBUG_LAYOUT:Z
  
      if-eqz v31, :cond_28
***************
*** 42743,42748 ****
--- 42781,42788 ----
  
      invoke-static/range {v18 .. v19}, Landroid/os/Binder;->restoreCallingIdentity(J)V
  
+     invoke-static/range {v29 .. v29}, Lcom/android/server/wm/WindowManagerService$Injector;->saveFloatingWindowState(Lcom/android/server/wm/WindowState;)V
+ 
      const/16 v31, 0x0
  
      monitor-exit v32
***************
*** 50084,50089 ****
--- 50124,50140 ----
      goto :goto_1
  .end method
  
+ .method public shutdown()V
+     .locals 1
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     invoke-virtual {p0, v0}, Lcom/android/server/wm/WindowManagerService;->shutdown(Z)V
+ 
+     return-void
+ .end method
+ 
  .method public shutdown(Z)V
      .locals 1
      .parameter "confirm"
