*** ./smali/com/android/server/net/NetworkPolicyManagerService.smali	2014-10-11 10:21:21.468209727 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/services.jar.out/./smali/com/android/server/net/NetworkPolicyManagerService.smali	2014-10-11 10:18:20.412205800 +1100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/net/NetworkPolicyManagerService$XmlUtils;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/server/net/NetworkPolicyManagerService$XmlUtils;,
!         Lcom/android/server/net/NetworkPolicyManagerService$Injector;
      }
  .end annotation
  
***************
*** 1087,1093 ****
  .end method
  
  .method private static buildViewDataUsageIntent(Landroid/net/NetworkTemplate;)Landroid/content/Intent;
!     .locals 4
      .parameter "template"
  
      .prologue
--- 1088,1094 ----
  .end method
  
  .method private static buildViewDataUsageIntent(Landroid/net/NetworkTemplate;)Landroid/content/Intent;
!     .locals 2
      .parameter "template"
  
      .prologue
***************
*** 1096,1110 ****
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
      .local v0, intent:Landroid/content/Intent;
!     new-instance v1, Landroid/content/ComponentName;
  
!     const-string v2, "com.android.settings"
! 
!     const-string v3, "com.android.settings.Settings$DataUsageSummaryActivity"
! 
!     invoke-direct {v1, v2, v3}, Landroid/content/ComponentName;-><init>(Ljava/lang/String;Ljava/lang/String;)V
! 
!     invoke-virtual {v0, v1}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
      const v1, 0x10008000
  
--- 1097,1105 ----
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v1, "android.intent.action.VIEW_DATA_USAGE_SUMMARY"
  
!     invoke-virtual {v0, v1}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
      const v1, 0x10008000
  
***************
*** 1571,1580 ****
  
      .restart local v15       #title:Ljava/lang/CharSequence;
      :goto_2
-     const/4 v1, 0x1
- 
-     invoke-virtual {v9, v1}, Landroid/app/Notification$Builder;->setOngoing(Z)Landroid/app/Notification$Builder;
- 
      const v1, 0x1080530
  
      invoke-virtual {v9, v1}, Landroid/app/Notification$Builder;->setSmallIcon(I)Landroid/app/Notification$Builder;
--- 1566,1571 ----
***************
*** 1702,1711 ****
  
      .restart local v15       #title:Ljava/lang/CharSequence;
      :goto_3
-     const/4 v1, 0x1
- 
-     invoke-virtual {v9, v1}, Landroid/app/Notification$Builder;->setOngoing(Z)Landroid/app/Notification$Builder;
- 
      const v1, 0x1080078
  
      invoke-virtual {v9, v1}, Landroid/app/Notification$Builder;->setSmallIcon(I)Landroid/app/Notification$Builder;
--- 1693,1698 ----
***************
*** 1928,1933 ****
--- 1915,1953 ----
      goto :goto_0
  .end method
  
+ .method private enqueueValidNotification(Landroid/net/NetworkPolicy;IJ)V
+     .locals 2
+     .parameter "policy"
+     .parameter "type"
+     .parameter "totalBytes"
+ 
+     .prologue
+     invoke-static {p2}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->isIntervalValid(I)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     invoke-static {p2}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->setInterval(I)V
+ 
+     invoke-direct {p0, p1, p2, p3, p4}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_0
+     invoke-direct {p0, p1, p2}, Lcom/android/server/net/NetworkPolicyManagerService;->buildNotificationTag(Landroid/net/NetworkPolicy;I)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     .local v0, tag:Ljava/lang/String;
+     iget-object v1, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mActiveNotifs:Ljava/util/HashSet;
+ 
+     invoke-virtual {v1, v0}, Ljava/util/HashSet;->add(Ljava/lang/Object;)Z
+ 
+     goto :goto_0
+ .end method
+ 
  .method private ensureActiveMobilePolicyLocked()V
      .locals 15
  
***************
*** 2735,2741 ****
  .end method
  
  .method private getTotalBytes(Landroid/net/NetworkTemplate;JJ)J
!     .locals 9
      .parameter "template"
      .parameter "start"
      .parameter "end"
--- 2755,2761 ----
  .end method
  
  .method private getTotalBytes(Landroid/net/NetworkTemplate;JJ)J
!     .locals 10
      .parameter "template"
      .parameter "start"
      .parameter "end"
***************
*** 2766,2780 ****
  
      invoke-static {v0, v1}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
!     iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mNetworkStats:Landroid/net/INetworkStatsService;
  
!     move-object v1, p1
  
!     move-wide v2, p2
  
!     move-wide v4, p4
  
!     invoke-interface/range {v0 .. v5}, Landroid/net/INetworkStatsService;->getNetworkTotalBytes(Landroid/net/NetworkTemplate;JJ)J
  
      move-result-wide v7
  
--- 2786,2802 ----
  
      invoke-static {v0, v1}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
!     iget-object v1, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mNetworkStats:Landroid/net/INetworkStatsService;
  
!     move-object v0, p0
! 
!     move-object v2, p1
  
!     move-wide v3, p2
  
!     move-wide v5, p4
  
!     invoke-static/range {v0 .. v6}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->getNetworkTotalBytes(Lcom/android/server/net/NetworkPolicyManagerService;Landroid/net/INetworkStatsService;Landroid/net/NetworkTemplate;JJ)J
  
      move-result-wide v7
  
***************
*** 4472,4477 ****
--- 4494,4501 ----
      .prologue
      const/4 v3, -0x1
  
+     invoke-static {p0, p1, p2}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->setNetworkTemplateEnabled(Lcom/android/server/net/NetworkPolicyManagerService;Landroid/net/NetworkTemplate;Z)V
+ 
      iget-object v2, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mContext:Landroid/content/Context;
  
      invoke-static {v2}, Landroid/telephony/TelephonyManager;->from(Landroid/content/Context;)Landroid/telephony/TelephonyManager;
***************
*** 5206,5211 ****
--- 5230,5243 ----
  
      invoke-static {v2, v3}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     iget-object v2, v14, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
+ 
+     const/4 v3, 0x1
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-static {v0, v2, v3}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->setNetworkTemplateEnabled(Lcom/android/server/net/NetworkPolicyManagerService;Landroid/net/NetworkTemplate;Z)V
+ 
      goto :goto_0
  
      :cond_2
***************
*** 5387,5392 ****
--- 5419,5430 ----
  
      invoke-static {v2, v3}, Landroid/util/Slog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
+     iget-object v2, v14, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
+ 
+     move-object/from16 v0, p0
+ 
+     invoke-static {v0, v2, v12}, Lcom/android/server/net/NetworkPolicyManagerService$Injector;->setNetworkTemplateEnabled(Lcom/android/server/net/NetworkPolicyManagerService;Landroid/net/NetworkTemplate;Z)V
+ 
      goto/16 :goto_0
  
      .end local v12           #networkEnabled:Z
***************
*** 6640,6646 ****
  
      const/4 v0, 0x3
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto/16 :goto_0
  
--- 6678,6684 ----
  
      const/4 v0, 0x3
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueValidNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto/16 :goto_0
  
***************
*** 6653,6659 ****
  
      const/4 v0, 0x2
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      iget-object v0, v10, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
  
--- 6691,6697 ----
  
      const/4 v0, 0x2
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueValidNotification(Landroid/net/NetworkPolicy;IJ)V
  
      iget-object v0, v10, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
  
***************
*** 6680,6686 ****
  
      const/4 v0, 0x1
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto/16 :goto_0
  
--- 6718,6724 ----
  
      const/4 v0, 0x1
  
!     invoke-direct {p0, v10, v0, v12, v13}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueValidNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto/16 :goto_0
  
***************
*** 8253,8258 ****
--- 8291,8305 ----
      goto/16 :goto_2
  .end method
  
+ .method getContext()Landroid/content/Context;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mContext:Landroid/content/Context;
+ 
+     return-object v0
+ .end method
+ 
  .method public getNetworkPolicies()[Landroid/net/NetworkPolicy;
      .locals 3
  
