*** ./smali/com/android/internal/view/menu/ActionMenuView.smali	2014-10-11 10:20:10.456208187 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/view/menu/ActionMenuView.smali	2014-10-11 10:17:08.408204238 +1100
***************
*** 11,17 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/view/menu/ActionMenuView$LayoutParams;,
!         Lcom/android/internal/view/menu/ActionMenuView$ActionMenuChildView;
      }
  .end annotation
  
--- 11,18 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Lcom/android/internal/view/menu/ActionMenuView$LayoutParams;,
!         Lcom/android/internal/view/menu/ActionMenuView$ActionMenuChildView;,
!         Lcom/android/internal/view/menu/ActionMenuView$Injector;
      }
  .end annotation
  
***************
*** 2224,2229 ****
--- 2225,2242 ----
  
      invoke-virtual {v0, v12, v1, v2, v3}, Landroid/view/View;->layout(IIII)V
  
+     move-object/from16 v0, v29
+ 
+     move/from16 v1, p2
+ 
+     move/from16 v2, v26
+ 
+     move/from16 v3, p4
+ 
+     move/from16 v4, p5
+ 
+     invoke-static {v0, v1, v2, v3, v4}, Lcom/android/internal/view/menu/ActionMenuView$Injector;->layout(Landroid/view/View;IIII)V
+ 
      goto/16 :goto_0
  
      .end local v9           #height:I
