*** ./smali/android/content/res/Configuration.smali	2014-10-11 10:20:33.560208688 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/content/res/Configuration.smali	2014-10-11 10:17:29.964204706 +1100
***************
*** 156,161 ****
--- 156,169 ----
  
  .field public static final UI_MODE_TYPE_NORMAL:I = 0x1
  
+ .field public static final UI_MODE_TYPE_SCALE_HUGE:I = 0xf
+ 
+ .field public static final UI_MODE_TYPE_SCALE_LARGE:I = 0xe
+ 
+ .field public static final UI_MODE_TYPE_SCALE_MEDIUM:I = 0xd
+ 
+ .field public static final UI_MODE_TYPE_SCALE_SMALL:I = 0xc
+ 
  .field public static final UI_MODE_TYPE_TELEVISION:I = 0x4
  
  .field public static final UI_MODE_TYPE_UNDEFINED:I
***************
*** 170,175 ****
--- 178,185 ----
  
  .field public densityDpi:I
  
+ .field public extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
  .field public fontScale:F
  
  .field public hardKeyboardHidden:I
***************
*** 232,266 ****
  .end method
  
  .method public constructor <init>()V
!     .locals 0
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 0
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 0
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
--- 242,294 ----
  .end method
  
  .method public constructor <init>()V
!     .locals 1
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 1
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
***************
*** 315,328 ****
  
      and-int/2addr v0, p0
  
!     if-eqz v0, :cond_0
  
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_0
      const/4 v0, 0x0
  
      goto :goto_0
--- 343,363 ----
  
      and-int/2addr v0, p0
  
!     if-nez v0, :cond_0
! 
!     invoke-static {p0}, Lmiui/content/res/ExtraConfiguration;->needNewResources(I)Z
! 
!     move-result v0
  
+     if-eqz v0, :cond_1
+ 
+     :cond_0
      const/4 v0, 0x1
  
      :goto_0
      return v0
  
!     :cond_1
      const/4 v0, 0x0
  
      goto :goto_0
***************
*** 702,707 ****
--- 737,750 ----
  
      sub-int v2, v5, v6
  
+     iget-object v5, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v6, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v5, v6}, Lmiui/content/res/ExtraConfiguration;->compareTo(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v2
+ 
      if-nez v2, :cond_0
  
      iget-object v5, p0, Landroid/content/res/Configuration;->skin:Ljava/lang/String;
***************
*** 1045,1050 ****
--- 1088,1103 ----
      or-int/2addr v0, v1
  
      :cond_13
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->diff(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  .end method
  
***************
*** 1257,1263 ****
      :cond_0
      add-int v0, v1, v2
  
!     return v0
  
      :cond_1
      move v1, v2
--- 1310,1324 ----
      :cond_0
      add-int v0, v1, v2
  
!     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
! 
!     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->hashCode()I
! 
!     move-result v1
! 
!     add-int/2addr v1, v0
! 
!     return v1
  
      :cond_1
      move v1, v2
***************
*** 1531,1536 ****
--- 1592,1601 ----
  
      iput-object v0, p0, Landroid/content/res/Configuration;->skin:Ljava/lang/String;
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1}, Lmiui/content/res/ExtraConfiguration;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  
      :cond_1
***************
*** 1588,1594 ****
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
--- 1653,1659 ----
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 2
      .parameter "o"
  
      .prologue
***************
*** 1699,1704 ****
--- 1764,1775 ----
  
      iput-object v0, p0, Landroid/content/res/Configuration;->skin:Ljava/lang/String;
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v1, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, v1}, Lmiui/content/res/ExtraConfiguration;->setTo(Lmiui/content/res/ExtraConfiguration;)V
+ 
      return-void
  .end method
  
***************
*** 1717,1722 ****
--- 1788,1795 ----
      iput v1, p0, Landroid/content/res/Configuration;->mnc:I
  
      iput v1, p0, Landroid/content/res/Configuration;->mcc:I
+     
+     invoke-direct {p0}, Landroid/content/res/Configuration;->loadPreOperator()V
  
      iput-object v2, p0, Landroid/content/res/Configuration;->locale:Ljava/util/Locale;
  
***************
*** 1760,1765 ****
--- 1833,1842 ----
  
      iput-object v2, p0, Landroid/content/res/Configuration;->skin:Ljava/lang/String;
  
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->setToDefaults()V
+ 
      return-void
  .end method
  
***************
*** 2092,2097 ****
--- 2169,2182 ----
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
      :cond_0
+     iget-object v2, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v2}, Lmiui/content/res/ExtraConfiguration;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      const/16 v2, 0x7d
  
      invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
***************
*** 3030,3035 ****
--- 3115,3130 ----
      iput-object v1, p0, Landroid/content/res/Configuration;->skin:Ljava/lang/String;
  
      :cond_1c
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->updateFrom(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  
      :cond_1d
***************
*** 3160,3165 ****
--- 3255,3264 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1, p2}, Lmiui/content/res/ExtraConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
***************
*** 3201,3203 ****
--- 3300,3351 ----
  
      goto :goto_2
  .end method
+ 
+ .method private loadPreOperator()V
+     .locals 3
+ 
+     .prologue
+     const/16 v2, 0x1cc
+ 
+     const-string v1, "ro.carrier.name"
+ 
+     invoke-static {v1}, Landroid/os/SystemProperties;->get(Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     .local v0, carrier:Ljava/lang/String;
+     const-string v1, "ct"
+ 
+     invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_1
+ 
+     iput v2, p0, Landroid/content/res/Configuration;->mcc:I
+ 
+     const/4 v1, 0x3
+ 
+     iput v1, p0, Landroid/content/res/Configuration;->mnc:I
+ 
+     :cond_0
+     :goto_0
+     return-void
+ 
+     :cond_1
+     const-string v1, "cu"
+ 
+     invoke-virtual {v1, v0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iput v2, p0, Landroid/content/res/Configuration;->mcc:I
+ 
+     const/4 v1, 0x1
+ 
+     iput v1, p0, Landroid/content/res/Configuration;->mnc:I
+ 
+     goto :goto_0
+ .end method
