*** ./smali/android/content/res/Resources.smali	2014-10-11 10:20:33.576208688 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/content/res/Resources.smali	2014-10-11 10:17:29.984204706 +1100
***************
*** 7,13 ****
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/content/res/Resources$Theme;,
!         Landroid/content/res/Resources$NotFoundException;
      }
  .end annotation
  
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/content/res/Resources$Theme;,
!         Landroid/content/res/Resources$NotFoundException;,
!         Landroid/content/res/Resources$Injector;
      }
  .end annotation
  
***************
*** 200,206 ****
      return-void
  .end method
  
! .method private constructor <init>()V
      .locals 3
  
      .prologue
--- 201,207 ----
      return-void
  .end method
  
! .method constructor <init>()V
      .locals 3
  
      .prologue
***************
*** 560,565 ****
--- 561,585 ----
      return-void
  .end method
  
+ .method static clearPreloadedCache()V
+     .locals 1
+ 
+     .prologue
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorStateLists:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     return-void
+ .end method
+ 
  .method private generateCurrentDensitySuffix()Ljava/lang/String;
      .locals 8
  
***************
*** 864,870 ****
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
!     new-instance v0, Landroid/content/res/TypedArray;
  
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      mul-int/lit8 v2, p1, 0x6
--- 884,890 ----
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
!     new-instance v0, Landroid/content/res/MiuiTypedArray;
  
      .end local v0           #attrs:Landroid/content/res/TypedArray;
      mul-int/lit8 v2, p1, 0x6
***************
*** 875,881 ****
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/TypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
--- 895,901 ----
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/MiuiTypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
***************
*** 925,931 ****
  .end method
  
  .method public static getSystem()Landroid/content/res/Resources;
!     .locals 3
  
      .prologue
      sget-object v2, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
--- 945,951 ----
  .end method
  
  .method public static getSystem()Landroid/content/res/Resources;
!     .locals 5
  
      .prologue
      sget-object v2, Landroid/content/res/Resources;->mSync:Ljava/lang/Object;
***************
*** 938,951 ****
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/Resources;
  
      .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/Resources;-><init>()V
  
      .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      :cond_0
      monitor-exit v2
  
--- 958,979 ----
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/MiuiResources;
  
      .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/MiuiResources;-><init>()V
  
      .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
+     move-object v1, v0
+ 
+     check-cast v1, Landroid/content/res/MiuiResources;
+ 
+     const/4 v4, 0x0
+ 
+     invoke-virtual {v1, v4}, Landroid/content/res/MiuiResources;->init(Ljava/lang/String;)V
+ 
      :cond_0
      monitor-exit v2
  
***************
*** 1841,1846 ****
--- 1869,1876 ----
      .end annotation
  
      .prologue
+     goto :goto_miui_0
+     
      sget-object v1, Landroid/content/res/Resources;->sPreloadedColorStateLists:Landroid/util/LongSparseArray;
  
      invoke-virtual {v1}, Landroid/util/LongSparseArray;->size()I
***************
*** 1862,1867 ****
--- 1892,1898 ----
      invoke-direct {p0}, Landroid/content/res/Resources;->loadPreloadedColorResources()V
  
      :cond_0
+     :goto_miui_0
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
***************
*** 2246,2251 ****
--- 2277,2284 ----
      .end annotation
  
      .prologue
+     goto :goto_miui_0
+     
      sget-object v1, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
  
      invoke-virtual {v1}, Landroid/util/LongSparseArray;->size()I
***************
*** 2267,2272 ****
--- 2300,2306 ----
      invoke-direct {p0}, Landroid/content/res/Resources;->loadPreloadedDrawableResources()V
  
      :cond_0
+     :goto_miui_0
      iget-object v2, p0, Landroid/content/res/Resources;->mTmpValue:Landroid/util/TypedValue;
  
      monitor-enter v2
***************
*** 2713,2718 ****
--- 2747,2781 ----
      goto :goto_0
  .end method
  
+ .method getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
+     .locals 2
+     .parameter "key"
+     .parameter "id"
+ 
+     .prologue
+     sget-object v1, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v1, p1, p2}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
+ 
+     move-result-object v0
+ 
+     check-cast v0, Landroid/graphics/drawable/Drawable$ConstantState;
+ 
+     .local v0, cs:Landroid/graphics/drawable/Drawable$ConstantState;
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0, p3}, Landroid/content/res/Resources;->isPreloadOverlayed(I)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     const/4 v0, 0x0
+ 
+     :cond_0
+     return-object v0
+ .end method
+ 
  .method public getQuantityString(II)Ljava/lang/String;
      .locals 1
      .parameter "id"
***************
*** 3971,3976 ****
--- 4034,4058 ----
      return-object v0
  .end method
  
+ .method isPreloadOverlayed(I)Z
+     .locals 1
+     .parameter "id"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
+ .method isPreloading()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/content/res/Resources;->mPreloading:Z
+ 
+     return v0
+ .end method
+ 
  .method loadColorStateList(Landroid/util/TypedValue;I)Landroid/content/res/ColorStateList;
      .locals 12
      .parameter "value"
***************
*** 4493,4514 ****
  
      if-ne v13, v14, :cond_7
  
!     sget-object v13, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
! 
!     invoke-virtual {v13, v9, v10}, Landroid/util/LongSparseArray;->get(J)Ljava/lang/Object;
  
!     move-result-object v13
  
!     check-cast v13, Landroid/graphics/drawable/Drawable$ConstantState;
  
!     move-object v2, v13
  
      goto :goto_3
  
      :cond_7
!     const/4 v13, 0x0
! 
!     move-object v2, v13
  
      goto :goto_3
  
--- 4575,4592 ----
  
      if-ne v13, v14, :cond_7
  
!     move-object/from16 v0, p0
  
!     move/from16 v1, p2
  
!     invoke-virtual {v0, v9, v10, v1}, Landroid/content/res/Resources;->getPreloadedDrawable(JI)Landroid/graphics/drawable/Drawable$ConstantState;
  
!     move-result-object v2
  
      goto :goto_3
  
      :cond_7
!     const/4 v2, 0x0
  
      goto :goto_3
  
***************
*** 4672,4684 ****
      move-result-object v7
  
      .local v7, is:Ljava/io/InputStream;
      const/4 v13, 0x0
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, p1
  
!     invoke-static {v0, v1, v7, v6, v13}, Landroid/graphics/drawable/Drawable;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
--- 4750,4764 ----
      move-result-object v7
  
      .local v7, is:Ljava/io/InputStream;
+     invoke-static/range {p2 .. p2}, Landroid/content/res/Resources$Injector;->setDrawableId(I)V
+ 
      const/4 v13, 0x0
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, p1
  
!     invoke-static {v0, v1, v7, v6, v13}, Landroid/content/res/Resources$Injector;->createFromResourceStream(Landroid/content/res/Resources;Landroid/util/TypedValue;Ljava/io/InputStream;Ljava/lang/String;Landroid/graphics/BitmapFactory$Options;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v3
  
***************
*** 4795,4800 ****
--- 4875,4899 ----
      goto :goto_6
  .end method
  
+ .method loadOverlayDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
+     .locals 1
+     .parameter "value"
+     .parameter "id"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
+ .method loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
+     .locals 0
+     .parameter "array"
+ 
+     .prologue
+     return-object p1
+ .end method
+ 
  .method loadXmlResourceParser(ILjava/lang/String;)Landroid/content/res/XmlResourceParser;
      .locals 5
      .parameter "id"
***************
*** 5188,5194 ****
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     return-object v0
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
--- 5287,5297 ----
  
      iput-object v2, v0, Landroid/content/res/TypedArray;->mXml:Landroid/content/res/XmlBlock$Parser;
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v3
! 
!     return-object v3
  .end method
  
  .method public obtainTypedArray(I)Landroid/content/res/TypedArray;
***************
*** 5260,5266 ****
  
      aput v4, v2, v4
  
!     return-object v0
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
--- 5363,5373 ----
  
      aput v4, v2, v4
  
!     invoke-virtual {p0, v0}, Landroid/content/res/Resources;->loadOverlayTypedArray(Landroid/content/res/TypedArray;)Landroid/content/res/TypedArray;
! 
!     move-result-object v2
! 
!     return-object v2
  .end method
  
  .method public openRawResource(I)Ljava/io/InputStream;
***************
*** 6168,6174 ****
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v20
  
      .end local v21           #density:I
      :cond_a
--- 6275,6287 ----
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v2
! 
!     const/high16 v3, -0x8000
! 
!     and-int v3, v3, v20
! 
!     or-int v20, v2, v3
  
      .end local v21           #density:I
      :cond_a
