*** ./smali/android/widget/AbsListView$FlingRunnable.smali	2014-10-11 10:21:11.320209507 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/widget/AbsListView$FlingRunnable.smali	2014-10-11 10:18:09.676205567 +1100
***************
*** 70,75 ****
--- 70,126 ----
      return-object v0
  .end method
  
+ .method private endFling2()V
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     #getter for: Landroid/widget/AbsListView;->mScrollY:I
+     invoke-static {v0}, Landroid/widget/AbsListView;->access$3100(Landroid/widget/AbsListView;)I
+ 
+     move-result v0
+ 
+     if-eqz v0, :cond_0
+ 
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->startSpringback()V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_0
+     invoke-virtual {p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
+ 
+     goto :goto_0
+ .end method
+ 
+ .method private scaleOverScrollDeltaY(II)I
+     .locals 2
+     .parameter "deltaY"
+     .parameter "scrollY"
+ 
+     .prologue
+     move v0, p1
+ 
+     .local v0, retval:I
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     invoke-virtual {v1}, Landroid/widget/OverScroller;->checkSpringBackState()Z
+ 
+     move-result v1
+ 
+     if-nez v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
+ 
+     invoke-static {v1, p1, p2}, Landroid/widget/AbsListView$Injector;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
+ 
+     move-result v0
+ 
+     :cond_0
+     return v0
+ .end method
+ 
  
  # virtual methods
  .method edgeReached(I)V
***************
*** 459,464 ****
--- 510,524 ----
      return-void
  .end method
  
+ .method protected getScroller()Landroid/widget/OverScroller;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/AbsListView$FlingRunnable;->mScroller:Landroid/widget/OverScroller;
+ 
+     return-object v0
+ .end method
+ 
  .method public run()V
      .locals 30
  
***************
*** 907,913 ****
  
      sub-int v2, v20, v2
  
!     neg-int v4, v2
  
      .local v4, overshoot:I
      move-object/from16 v0, p0
--- 967,989 ----
  
      sub-int v2, v20, v2
  
!     neg-int v2, v2
! 
!     move-object/from16 v0, p0
! 
!     iget-object v3, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
! 
!     move-object/from16 v0, p0
! 
!     iget-object v5, v0, Landroid/widget/AbsListView$FlingRunnable;->this$0:Landroid/widget/AbsListView;
! 
!     invoke-virtual {v5}, Landroid/widget/AbsListView;->getScrollY()I
! 
!     move-result v5
! 
!     invoke-static {v3, v2, v5}, Landroid/widget/AbsListView$Injector;->scaleOverScrollDeltaY(Landroid/widget/AbsListView;II)I
! 
!     move-result v4
  
      .local v4, overshoot:I
      move-object/from16 v0, p0
***************
*** 1132,1138 ****
      move-result v19
  
      .local v19, currY:I
!     sub-int v7, v19, v9
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
--- 1208,1220 ----
      move-result v19
  
      .local v19, currY:I
!     sub-int v2, v19, v9
! 
!     move-object/from16 v0, p0
! 
!     invoke-direct {v0, v2, v9}, Landroid/widget/AbsListView$FlingRunnable;->scaleOverScrollDeltaY(II)I
! 
!     move-result v7
  
      .local v7, deltaY:I
      move-object/from16 v0, p0
***************
*** 1442,1448 ****
      invoke-static {v2, v3}, Lcom/mediatek/xlog/Xlog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      :cond_16
!     invoke-virtual/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling()V
  
      goto/16 :goto_0
  
--- 1524,1530 ----
      invoke-static {v2, v3}, Lcom/mediatek/xlog/Xlog;->d(Ljava/lang/String;Ljava/lang/String;)I
  
      :cond_16
!     invoke-direct/range {p0 .. p0}, Landroid/widget/AbsListView$FlingRunnable;->endFling2()V
  
      goto/16 :goto_0
  
