*** ./smali/android/widget/CompoundButton.smali	2014-10-11 10:21:10.848209497 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/widget/CompoundButton.smali	2014-10-11 10:18:09.180205556 +1100
***************
*** 11,17 ****
      value = {
          Landroid/widget/CompoundButton$1;,
          Landroid/widget/CompoundButton$SavedState;,
!         Landroid/widget/CompoundButton$OnCheckedChangeListener;
      }
  .end annotation
  
--- 11,18 ----
      value = {
          Landroid/widget/CompoundButton$1;,
          Landroid/widget/CompoundButton$SavedState;,
!         Landroid/widget/CompoundButton$OnCheckedChangeListener;,
!         Landroid/widget/CompoundButton$Injector;
      }
  .end annotation
  
***************
*** 285,291 ****
      and-int/lit8 v7, v8, 0x70
  
      .local v7, verticalGravity:I
!     invoke-virtual {v1}, Landroid/graphics/drawable/Drawable;->getIntrinsicHeight()I
  
      move-result v2
  
--- 286,292 ----
      and-int/lit8 v7, v8, 0x70
  
      .local v7, verticalGravity:I
!     invoke-static {v1, p0}, Landroid/widget/CompoundButton$Injector;->getDrawableHeight(Landroid/graphics/drawable/Drawable;Landroid/widget/CompoundButton;)I
  
      move-result v2
  
