*** ./smali/android/widget/MediaController.smali	2014-10-11 10:21:08.812209453 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/widget/MediaController.smali	2014-10-11 10:18:07.552205521 +1100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/widget/MediaController$MediaPlayerControl;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Landroid/widget/MediaController$MediaPlayerControl;,
!         Landroid/widget/MediaController$Injector;
      }
  .end annotation
  
***************
*** 1713,1718 ****
--- 1714,1730 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageButton;->setImageResource(I)V
  
+     :goto_miui_0
+     iget-object v0, p0, Landroid/widget/MediaController;->mPlayer:Landroid/widget/MediaController$MediaPlayerControl;
+ 
+     invoke-interface {v0}, Landroid/widget/MediaController$MediaPlayerControl;->isPlaying()Z
+ 
+     move-result v0
+ 
+     iget-object v1, p0, Landroid/widget/MediaController;->mPauseButton:Landroid/widget/ImageButton;
+ 
+     invoke-static {v0, v1}, Landroid/widget/MediaController$Injector;->updatePausePlay(ZLandroid/widget/ImageView;)V
+ 
      goto :goto_0
  
      :cond_2
***************
*** 1722,1726 ****
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageButton;->setImageResource(I)V
  
!     goto :goto_0
  .end method
--- 1734,1738 ----
  
      invoke-virtual {v0, v1}, Landroid/widget/ImageButton;->setImageResource(I)V
  
!     goto :goto_miui_0
  .end method
