*** ./smali/android/widget/OverScroller.smali	2014-10-11 10:21:11.004209500 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/widget/OverScroller.smali	2014-10-11 10:18:09.332205560 +1100
***************
*** 30,35 ****
--- 30,41 ----
  
  .field private final mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
+ .field mSpringOverScrollEnable:Z
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_FIELD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ .end field
+ 
  
  # direct methods
  .method public constructor <init>(Landroid/content/Context;)V
***************
*** 131,136 ****
--- 137,176 ----
      return-void
  .end method
  
+ .method public final checkSpringBackState()Z
+     .locals 2
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mState:I
+     invoke-static {v1}, Landroid/widget/OverScroller$SplineOverScroller;->access$700(Landroid/widget/OverScroller$SplineOverScroller;)I
+ 
+     move-result v1
+ 
+     if-eq v1, v0, :cond_0
+ 
+     iget-object v1, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mState:I
+     invoke-static {v1}, Landroid/widget/OverScroller$SplineOverScroller;->access$700(Landroid/widget/OverScroller$SplineOverScroller;)I
+ 
+     move-result v1
+ 
+     if-ne v1, v0, :cond_1
+ 
+     :cond_0
+     :goto_0
+     return v0
+ 
+     :cond_1
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public computeScrollOffset()Z
      .locals 8
  
***************
*** 243,249 ****
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
!     invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->update()Z
  
      move-result v6
  
--- 283,291 ----
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
!     const/4 v7, 0x0
!     
!     invoke-virtual {v6, v7}, Landroid/widget/OverScroller$SplineOverScroller;->update(Z)Z
  
      move-result v6
  
***************
*** 251,257 ****
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
!     invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished()Z
  
      move-result v6
  
--- 293,301 ----
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
  
!     iget-boolean v7, p0, Landroid/widget/OverScroller;->mSpringOverScrollEnable:Z
! 
!     invoke-virtual {v6, v7}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished(Z)Z
  
      move-result v6
  
***************
*** 273,279 ****
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
!     invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->update()Z
  
      move-result v6
  
--- 317,325 ----
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
!     const/4 v7, 0x0
!     
!     invoke-virtual {v6, v7}, Landroid/widget/OverScroller$SplineOverScroller;->update(Z)Z
  
      move-result v6
  
***************
*** 281,287 ****
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
!     invoke-virtual {v6}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished()Z
  
      move-result v6
  
--- 327,335 ----
  
      iget-object v6, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
  
!     iget-boolean v7, p0, Landroid/widget/OverScroller;->mSpringOverScrollEnable:Z
! 
!     invoke-virtual {v6, v7}, Landroid/widget/OverScroller$SplineOverScroller;->continueWhenFinished(Z)Z
  
      move-result v6
  
***************
*** 571,576 ****
--- 619,641 ----
      return v0
  .end method
  
+ .method public getCurrYVelocity()F
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     #getter for: Landroid/widget/OverScroller$SplineOverScroller;->mCurrVelocity:F
+     invoke-static {v0}, Landroid/widget/OverScroller$SplineOverScroller;->access$200(Landroid/widget/OverScroller$SplineOverScroller;)F
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public final getDuration()I
      .locals 2
      .annotation runtime Ljava/lang/Deprecated;
***************
*** 853,858 ****
--- 918,942 ----
      return-void
  .end method
  
+ .method public final setDecelerationScale(F)V
+     .locals 1
+     .parameter "scale"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setDecelerationScale(F)V
+ 
+     iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setDecelerationScale(F)V
+ 
+     return-void
+ .end method
+ 
  .method public setFinalX(I)V
      .locals 1
      .parameter "newX"
***************
*** 906,911 ****
--- 990,1027 ----
  
      return-void
  .end method
+ 
+ .method public setSpringOverScrollEnable(Z)V
+     .locals 0
+     .parameter "enable"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iput-boolean p1, p0, Landroid/widget/OverScroller;->mSpringOverScrollEnable:Z
+ 
+     return-void
+ .end method
+ 
+ .method public final setTotalOverDistance(I)V
+     .locals 1
+     .parameter "distance"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerX:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setTotalOverDistance(I)V
+ 
+     iget-object v0, p0, Landroid/widget/OverScroller;->mScrollerY:Landroid/widget/OverScroller$SplineOverScroller;
+ 
+     invoke-virtual {v0, p1}, Landroid/widget/OverScroller$SplineOverScroller;->setTotalOverDistance(I)V
+ 
+     return-void
+ .end method
  
  .method public springBack(IIIIII)Z
      .locals 4
