*** ./smali/android/widget/QuickContactBadge$QueryHandler.smali	2014-10-11 10:21:11.084209502 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/widget/QuickContactBadge$QueryHandler.smali	2014-10-11 10:18:09.416205561 +1100
***************
*** 70,75 ****
--- 70,81 ----
      #calls: Landroid/widget/QuickContactBadge;->onContactUriChanged()V
      invoke-static {v7}, Landroid/widget/QuickContactBadge;->access$100(Landroid/widget/QuickContactBadge;)V
  
+     iget-object v7, p0, Landroid/widget/QuickContactBadge$QueryHandler;->this$0:Landroid/widget/QuickContactBadge;
+ 
+     invoke-static {v7, v6, v5, v2}, Landroid/widget/QuickContactBadge$Injector;->showQuickContactForStranger(Landroid/widget/QuickContactBadge;ZLandroid/net/Uri;Landroid/net/Uri;)Landroid/net/Uri;
+ 
+     move-result-object v2
+ 
      if-eqz v6, :cond_4
  
      if-eqz v5, :cond_4
