*** ./smali/com/android/server/location/GpsLocationProvider.smali	2014-10-11 10:21:20.988209717 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/services.jar.out/./smali/com/android/server/location/GpsLocationProvider.smali	2014-10-11 10:18:19.916205789 +1100
***************
*** 12,18 ****
          Lcom/android/server/location/GpsLocationProvider$NetworkLocationListener;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
          Lcom/android/server/location/GpsLocationProvider$Listener;,
!         Lcom/android/server/location/GpsLocationProvider$GpsRequest;
      }
  .end annotation
  
--- 12,19 ----
          Lcom/android/server/location/GpsLocationProvider$NetworkLocationListener;,
          Lcom/android/server/location/GpsLocationProvider$ProviderHandler;,
          Lcom/android/server/location/GpsLocationProvider$Listener;,
!         Lcom/android/server/location/GpsLocationProvider$GpsRequest;,
!         Lcom/android/server/location/GpsLocationProvider$Injector;
      }
  .end annotation
  
***************
*** 4385,4390 ****
--- 4386,4393 ----
  
      invoke-virtual {v2, v6, v8}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Z)Landroid/content/Intent;
  
+     invoke-static {p0, v2}, Lcom/android/server/location/GpsLocationProvider$Injector;->appendUidExtra(Lcom/android/server/location/GpsLocationProvider;Landroid/content/Intent;)V
+ 
      iget-object v6, p0, Lcom/android/server/location/GpsLocationProvider;->mContext:Landroid/content/Context;
  
      sget-object v8, Landroid/os/UserHandle;->ALL:Landroid/os/UserHandle;
***************
*** 6354,6359 ****
--- 6357,6371 ----
      return-void
  .end method
  
+ .method getClientUids()[I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/server/location/GpsLocationProvider;->mClientUids:[I
+ 
+     return-object v0
+ .end method
+ 
  .method public getFileTime([J)I
      .locals 1
      .parameter "times"
***************
*** 6384,6389 ****
--- 6396,6410 ----
      return-object v0
  .end method
  
+ .method getNavigating()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Lcom/android/server/location/GpsLocationProvider;->mNavigating:Z
+ 
+     return v0
+ .end method
+ 
  .method public getNetInitiatedListener()Landroid/location/INetInitiatedListener;
      .locals 1
  
