*** ./smali/android/app/DownloadManager$Request.smali	2014-10-11 10:21:05.084209372 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/app/DownloadManager$Request.smali	2014-10-11 10:18:03.656205436 +1100
***************
*** 762,768 ****
      return-object p0
  .end method
  
! .method public setUserAgent(Ljava/lang/String;)Landroid/app/DownloadManager$Request;
      .locals 3
      .parameter "userAgent"
  
--- 762,768 ----
      return-object p0
  .end method
  
! .method public setUserAgent(Ljava/lang/String;)V
      .locals 3
      .parameter "userAgent"
  
***************
*** 791,797 ****
  
      invoke-static {v0, v1}, Lcom/mediatek/xlog/Xlog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
!     return-object p0
  .end method
  
  .method public setVisibleInDownloadsUi(Z)Landroid/app/DownloadManager$Request;
--- 791,797 ----
  
      invoke-static {v0, v1}, Lcom/mediatek/xlog/Xlog;->v(Ljava/lang/String;Ljava/lang/String;)I
  
!     return-void
  .end method
  
  .method public setVisibleInDownloadsUi(Z)Landroid/app/DownloadManager$Request;
