*** ./smali/android/app/ProgressDialog.smali	2014-10-11 10:21:05.148209373 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/secondary-framework.jar.out/./smali/android/app/ProgressDialog.smali	2014-10-11 10:18:03.724205438 +1100
***************
*** 3,8 ****
--- 3,16 ----
  .source "ProgressDialog.java"
  
  
+ # annotations
+ .annotation system Ldalvik/annotation/MemberClasses;
+     value = {
+         Landroid/app/ProgressDialog$Injector;
+     }
+ .end annotation
+ 
+ 
  # static fields
  .field public static final STYLE_HORIZONTAL:I = 0x1
  
***************
*** 641,646 ****
--- 649,660 ----
  
      invoke-virtual {p0, v2}, Landroid/app/ProgressDialog;->setView(Landroid/view/View;)V
  
+     iget-object v3, p0, Landroid/app/ProgressDialog;->mContext:Landroid/content/Context;
+ 
+     iget-object v4, p0, Landroid/app/ProgressDialog;->mMessageView:Landroid/widget/TextView;
+ 
+     invoke-static {p0, v3, v4}, Landroid/app/ProgressDialog$Injector;->usingV5Style(Landroid/app/ProgressDialog;Landroid/content/Context;Landroid/widget/TextView;)V
+ 
      goto :goto_0
  .end method
  
