*** ./smali/android/webkit/WebViewClassic.smali	2014-10-11 10:20:36.508208752 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/webkit/WebViewClassic.smali	2014-10-11 10:17:33.272204777 +1100
***************
*** 23198,23204 ****
      goto :goto_0
  .end method
  
! .method sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
--- 23198,23204 ----
      goto :goto_0
  .end method
  
! .method public sendOurVisibleRect()Landroid/graphics/Rect;
      .locals 4
  
      .prologue
***************
*** 24639,24644 ****
--- 24639,24675 ----
      return-void
  .end method
  
+ .method public setProxy(Ljava/lang/String;I)V
+     .locals 3
+     .parameter "host"
+     .parameter "port"
+ 
+     .prologue
+     const/4 v2, 0x0
+ 
+     const/16 v1, 0xc1
+ 
+     if-eqz p1, :cond_0
+ 
+     if-nez p2, :cond_1
+ 
+     :cond_0
+     invoke-static {v1, v2}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     :goto_0
+     return-void
+ 
+     :cond_1
+     new-instance v0, Landroid/net/ProxyProperties;
+ 
+     invoke-direct {v0, p1, p2, v2}, Landroid/net/ProxyProperties;-><init>(Ljava/lang/String;ILjava/lang/String;)V
+ 
+     .local v0, pp:Landroid/net/ProxyProperties;
+     invoke-static {v1, v0}, Landroid/webkit/WebViewCore;->sendStaticMessage(ILjava/lang/Object;)V
+ 
+     goto :goto_0
+ .end method
+ 
  .method public setScrollBarStyle(I)V
      .locals 1
      .parameter "style"
***************
*** 25717,25719 ****
--- 25748,25808 ----
  
      return v0
  .end method
+ 
+ .method public checkIfReadModeAvailable()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public checkIfReadModeAvailable(Z)V
+     .locals 0
+     .parameter "inSession"
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public enterReadMode()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public isHitSelectionHandle()Z
+    .locals 1
+ 
+    .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
+ 
+ .method public isSelectingText()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/webkit/WebViewClassic;->mSelectingText:Z
+ 
+     return v0
+ .end method
+ 
+ .method public hideFloatView()V
+     .locals 0
+ 
+     .prologue
+     return-void
+ .end method
+ 
+ .method public checkIfSaveImageFromCacheAvailable(Ljava/lang/String;Ljava/lang/String;)Z
+     .locals 1
+     .parameter "imageUrl"
+     .parameter "path"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return v0
+ .end method
