*** ./smali/android/net/wifi/WifiStateMachine.smali	2014-10-11 10:20:28.412208576 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiStateMachine.smali	2014-10-11 10:17:23.700204570 +1100
***************
*** 1542,1553 ****
  
      invoke-virtual {p0, v6, v7}, Landroid/net/wifi/WifiStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
  
-     iget-object v6, p0, Landroid/net/wifi/WifiStateMachine;->mCaptivePortalCheckState:Lcom/android/internal/util/State;
- 
-     iget-object v7, p0, Landroid/net/wifi/WifiStateMachine;->mL2ConnectedState:Lcom/android/internal/util/State;
- 
-     invoke-virtual {p0, v6, v7}, Landroid/net/wifi/WifiStateMachine;->addState(Lcom/android/internal/util/State;Lcom/android/internal/util/State;)V
- 
      iget-object v6, p0, Landroid/net/wifi/WifiStateMachine;->mConnectedState:Lcom/android/internal/util/State;
  
      iget-object v7, p0, Landroid/net/wifi/WifiStateMachine;->mL2ConnectedState:Lcom/android/internal/util/State;
--- 1542,1547 ----
***************
*** 6056,6061 ****
--- 6050,6059 ----
  
      invoke-virtual {v1, v4}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v1, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     invoke-virtual {v1, v2}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      const/4 v0, 0x3
  
      .local v0, maxRetries:I
***************
*** 6358,6363 ****
--- 6356,6365 ----
  
      invoke-virtual {v2, v7}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     invoke-virtual {v2, v7}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      sget-object v2, Landroid/net/NetworkInfo$DetailedState;->DISCONNECTED:Landroid/net/NetworkInfo$DetailedState;
  
      invoke-direct {p0, v2}, Landroid/net/wifi/WifiStateMachine;->setNetworkDetailedState(Landroid/net/NetworkInfo$DetailedState;)V
***************
*** 6614,6619 ****
--- 6616,6627 ----
  
      invoke-virtual {v7, v8}, Landroid/net/wifi/WifiInfo;->setMeteredHint(Z)V
  
+     iget-object v7, p0, Landroid/net/wifi/WifiStateMachine;->mWifiInfo:Landroid/net/wifi/WifiInfo;
+ 
+     iget-object v8, p1, Landroid/net/DhcpInfoInternal;->vendorInfo:Ljava/lang/String;
+ 
+     invoke-virtual {v7, v8}, Landroid/net/wifi/WifiInfo;->setVendorInfo(Ljava/lang/String;)V
+ 
      invoke-direct {p0}, Landroid/net/wifi/WifiStateMachine;->getNetworkDetailedState()Landroid/net/NetworkInfo$DetailedState;
  
      move-result-object v7
***************
*** 9051,9060 ****
  
      invoke-virtual {v0, v4}, Landroid/content/Intent;->setFlags(I)Landroid/content/Intent;
  
-     iget-object v4, p0, Landroid/net/wifi/WifiStateMachine;->mContext:Landroid/content/Context;
- 
-     invoke-virtual {v4, v0}, Landroid/content/Context;->startActivity(Landroid/content/Intent;)V
- 
      goto :goto_0
  .end method
  
--- 9059,9064 ----
***************
*** 11964,11966 ****
--- 11968,12071 ----
  
      throw v0
  .end method
+ 
+ .method private transitionConnectedState()V
+     .locals 5
+ 
+     .prologue
+     :try_start_0
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mNwService:Landroid/os/INetworkManagementService;
+ 
+     iget-object v3, p0, Landroid/net/wifi/WifiStateMachine;->mInterfaceName:Ljava/lang/String;
+ 
+     invoke-interface {v2, v3}, Landroid/os/INetworkManagementService;->enableIpv6(Ljava/lang/String;)V
+     :try_end_0
+     .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
+     .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_1
+ 
+     :goto_0
+     sget-object v2, Landroid/net/NetworkInfo$DetailedState;->CONNECTED:Landroid/net/NetworkInfo$DetailedState;
+ 
+     invoke-direct {p0, v2}, Landroid/net/wifi/WifiStateMachine;->setNetworkDetailedState(Landroid/net/NetworkInfo$DetailedState;)V
+ 
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mWifiConfigStore:Landroid/net/wifi/WifiConfigStore;
+ 
+     iget v3, p0, Landroid/net/wifi/WifiStateMachine;->mLastNetworkId:I
+ 
+     sget-object v4, Landroid/net/NetworkInfo$DetailedState;->CONNECTED:Landroid/net/NetworkInfo$DetailedState;
+ 
+     invoke-virtual {v2, v3, v4}, Landroid/net/wifi/WifiConfigStore;->updateStatus(ILandroid/net/NetworkInfo$DetailedState;)V
+ 
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mLastBssid:Ljava/lang/String;
+ 
+     invoke-direct {p0, v2}, Landroid/net/wifi/WifiStateMachine;->sendNetworkStateChangeBroadcast(Ljava/lang/String;)V
+ 
+     iget-object v2, p0, Landroid/net/wifi/WifiStateMachine;->mConnectedState:Lcom/android/internal/util/State;
+ 
+     invoke-virtual {p0, v2}, Landroid/net/wifi/WifiStateMachine;->transitionTo(Lcom/android/internal/util/IState;)V
+ 
+     return-void
+ 
+     :catch_0
+     move-exception v1
+ 
+     .local v1, re:Landroid/os/RemoteException;
+     new-instance v2, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v3, "Failed to enable IPv6: "
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-direct {p0, v2}, Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
+ 
+     goto :goto_0
+ 
+     .end local v1           #re:Landroid/os/RemoteException;
+     :catch_1
+     move-exception v0
+ 
+     .local v0, e:Ljava/lang/IllegalStateException;
+     new-instance v2, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v3, "Failed to enable IPv6: "
+ 
+     invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2, v0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-direct {p0, v2}, Landroid/net/wifi/WifiStateMachine;->loge(Ljava/lang/String;)V
+ 
+     goto :goto_0
+ .end method
+ 
+ .method static synthetic access$transitionConnectedState(Landroid/net/wifi/WifiStateMachine;)V
+     .locals 0
+     .parameter "x0"
+ 
+     .prologue
+     invoke-direct {p0}, Landroid/net/wifi/WifiStateMachine;->transitionConnectedState()V
+ 
+     return-void
+ .end method
