*** ./smali/android/net/wifi/WifiStateMachine$VerifyingLinkState.smali	2014-10-11 10:20:27.604208559 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiStateMachine$VerifyingLinkState.smali	2014-10-11 10:17:22.772204550 +1100
***************
*** 123,128 ****
--- 122,135 ----
      :pswitch_0
      iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$VerifyingLinkState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
+     invoke-static {v0}, Landroid/net/wifi/WifiStateMachine;->access$transitionConnectedState(Landroid/net/wifi/WifiStateMachine;)V
+ 
+     const/4 v0, 0x1
+ 
+     goto :goto_0
+ 
+     iget-object v0, p0, Landroid/net/wifi/WifiStateMachine$VerifyingLinkState;->this$0:Landroid/net/wifi/WifiStateMachine;
+ 
      iget-object v1, p0, Landroid/net/wifi/WifiStateMachine$VerifyingLinkState;->this$0:Landroid/net/wifi/WifiStateMachine;
  
      #getter for: Landroid/net/wifi/WifiStateMachine;->mCaptivePortalCheckState:Lcom/android/internal/util/State;
