*** ./smali/android/net/wifi/WifiInfo.smali	2014-10-11 10:20:29.316208596 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiInfo.smali	2014-10-11 10:17:24.736204592 +1100
***************
*** 58,63 ****
--- 58,65 ----
  
  .field private mWifiSsid:Landroid/net/wifi/WifiSsid;
  
+ .field private mVendorInfo:Ljava/lang/String;
+ 
  
  # direct methods
  .method static constructor <clinit>()V
***************
*** 271,276 ****
--- 273,282 ----
  
      iput-object v0, p0, Landroid/net/wifi/WifiInfo;->mIpV6Address:Ljava/net/InetAddress;
  
+     iget-object v0, p1, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     iput-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
      :cond_0
      return-void
  .end method
***************
*** 629,634 ****
--- 635,649 ----
      return-object v0
  .end method
  
+ .method public getVendorInfo()Ljava/lang/String;
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-object v0
+ .end method
+ 
  .method public getWifiSsid()Landroid/net/wifi/WifiSsid;
      .locals 1
  
***************
*** 766,771 ****
--- 781,796 ----
      return-void
  .end method
  
+ .method public setVendorInfo(Ljava/lang/String;)V
+     .locals 0
+     .parameter "vendorInfo"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-void
+ .end method
+ 
  .method public toString()Ljava/lang/String;
      .locals 4
  
***************
*** 997,1002 ****
--- 1022,1031 ----
      :goto_3
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeString(Ljava/lang/String;)V
+ 
      iget-object v0, p0, Landroid/net/wifi/WifiInfo;->mSupplicantState:Landroid/net/wifi/SupplicantState;
  
      invoke-virtual {v0, p1, p2}, Landroid/net/wifi/SupplicantState;->writeToParcel(Landroid/os/Parcel;I)V
***************
*** 1023,1025 ****
--- 1052,1065 ----
  
      goto :goto_3
  .end method
+ 
+ .method static synthetic access_setVendorInfo(Landroid/net/wifi/WifiInfo;Ljava/lang/String;)Ljava/lang/String;
+     .locals 0
+     .parameter "x0"
+     .parameter "x1"
+ 
+     .prologue
+     iput-object p1, p0, Landroid/net/wifi/WifiInfo;->mVendorInfo:Ljava/lang/String;
+ 
+     return-object p1
+ .end method
