*** ./smali/android/net/wifi/WifiManager.smali	2014-10-11 10:20:28.048208568 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/net/wifi/WifiManager.smali	2014-10-11 10:17:23.336204562 +1100
***************
*** 7,12 ****
--- 7,14 ----
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
          Landroid/net/wifi/WifiManager$1;,
+         Landroid/net/wifi/WifiManager$ChannelListener;,
+         Landroid/net/wifi/WifiManager$Channel;,
          Landroid/net/wifi/WifiManager$MulticastLock;,
          Landroid/net/wifi/WifiManager$WifiLock;,
          Landroid/net/wifi/WifiManager$ServiceHandler;,
***************
*** 754,759 ****
--- 756,772 ----
      return-void
  .end method
  
+ .method public cancelWps(Landroid/net/wifi/WifiManager$Channel;Landroid/net/wifi/WifiManager$ActionListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2}, Landroid/net/wifi/WifiManager;->cancelWps(Landroid/net/wifi/WifiManager$ActionListener;)V
+ 
+     return-void
+ .end method
+ 
  .method public captivePortalCheckComplete()V
      .locals 1
  
***************
*** 865,870 ****
--- 878,907 ----
      return-void
  .end method
  
+ .method public connect(Landroid/net/wifi/WifiManager$Channel;ILandroid/net/wifi/WifiManager$ActionListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "networkId"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/net/wifi/WifiManager;->connect(ILandroid/net/wifi/WifiManager$ActionListener;)V
+ 
+     return-void
+ .end method
+ 
+ .method public connect(Landroid/net/wifi/WifiManager$Channel;Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "config"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/net/wifi/WifiManager;->connect(Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+ 
+     return-void
+ .end method
+ 
  .method public createMulticastLock(Ljava/lang/String;)Landroid/net/wifi/WifiManager$MulticastLock;
      .locals 2
      .parameter "tag"
***************
*** 1173,1178 ****
--- 1210,1227 ----
      return-void
  .end method
  
+ .method public forget(Landroid/net/wifi/WifiManager$Channel;ILandroid/net/wifi/WifiManager$ActionListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "netId"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/net/wifi/WifiManager;->forget(ILandroid/net/wifi/WifiManager$ActionListener;)V
+ 
+     return-void
+ .end method
+ 
  .method public getAccessPointPreferredChannels()[Ljava/lang/String;
      .locals 2
  
***************
*** 1606,1611 ****
--- 1655,1672 ----
      goto :goto_0
  .end method
  
+ .method public initialize(Landroid/content/Context;Landroid/os/Looper;Landroid/net/wifi/WifiManager$ChannelListener;)Landroid/net/wifi/WifiManager$Channel;
+     .locals 1
+     .parameter "srcContext"
+     .parameter "srcLooper"
+     .parameter "listener"
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
+ 
  .method public initializeMulticastFiltering()Z
      .locals 2
  
***************
*** 1656,1661 ****
--- 1717,1756 ----
      goto :goto_0
  .end method
  
+ .method public isMiWifi()Z
+     .locals 3
+ 
+     .prologue
+     invoke-virtual {p0}, Landroid/net/wifi/WifiManager;->getConnectionInfo()Landroid/net/wifi/WifiInfo;
+ 
+     move-result-object v0
+ 
+     .local v0, wifiInfo:Landroid/net/wifi/WifiInfo;
+     if-eqz v0, :cond_0
+ 
+     const-string v1, "XIAOMI_ROUTER"
+ 
+     invoke-virtual {v0}, Landroid/net/wifi/WifiInfo;->getVendorInfo()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/String;->equalsIgnoreCase(Ljava/lang/String;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     const/4 v1, 0x1
+ 
+     :goto_0
+     return v1
+ 
+     :cond_0
+     const/4 v1, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method public isMulticastEnabled()Z
      .locals 2
  
***************
*** 1869,1874 ****
--- 1964,1981 ----
      return-void
  .end method
  
+ .method public save(Landroid/net/wifi/WifiManager$Channel;Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "config"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/net/wifi/WifiManager;->save(Landroid/net/wifi/WifiConfiguration;Landroid/net/wifi/WifiManager$ActionListener;)V
+ 
+     return-void
+ .end method
+ 
  .method public saveAPPriority()Z
      .locals 2
  
***************
*** 2225,2230 ****
--- 2332,2349 ----
      goto :goto_0
  .end method
  
+ .method public startWps(Landroid/net/wifi/WifiManager$Channel;Landroid/net/wifi/WpsInfo;Landroid/net/wifi/WifiManager$WpsListener;)V
+     .locals 0
+     .parameter "c"
+     .parameter "config"
+     .parameter "listener"
+ 
+     .prologue
+     invoke-virtual {p0, p2, p3}, Landroid/net/wifi/WifiManager;->startWps(Landroid/net/wifi/WpsInfo;Landroid/net/wifi/WifiManager$WpsListener;)V
+ 
+     return-void
+ .end method
+ 
  .method public startWps(Landroid/net/wifi/WpsInfo;Landroid/net/wifi/WifiManager$WpsListener;)V
      .locals 4
      .parameter "config"
