*** ./smali/android/graphics/drawable/DrawableContainer$DrawableContainerState.smali	2014-10-11 10:20:14.560208276 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/android/graphics/drawable/DrawableContainer$DrawableContainerState.smali	2014-10-11 10:17:11.872204313 +1100
***************
*** 206,211 ****
--- 206,219 ----
  
      iput v3, p0, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mConstantHeight:I
  
+     iget v3, p1, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mConstantMinimumWidth:I
+ 
+     iput v3, p0, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mConstantMinimumWidth:I
+ 
+     iget v3, p1, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mConstantMinimumHeight:I
+ 
+     iput v3, p0, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mConstantMinimumHeight:I
+ 
      iget v3, p1, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mOpacity:I
  
      iput v3, p0, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mOpacity:I
***************
*** 774,779 ****
--- 782,796 ----
      return v3
  .end method
  
+ .method public final getVariablePadding()Z
+     .locals 1
+ 
+     .prologue
+     iget-boolean v0, p0, Landroid/graphics/drawable/DrawableContainer$DrawableContainerState;->mVariablePadding:Z
+ 
+     return v0
+ .end method
+ 
  .method public growArray(II)V
      .locals 3
      .parameter "oldSize"
