*** ./smali/com/android/internal/telephony/DataConnectionTracker.smali	2014-10-11 10:20:00.864207979 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/DataConnectionTracker.smali	2014-10-11 10:17:01.432204087 +1100
***************
*** 11,17 ****
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
          Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;,
          Lcom/android/internal/telephony/DataConnectionTracker$Activity;,
!         Lcom/android/internal/telephony/DataConnectionTracker$State;
      }
  .end annotation
  
--- 11,18 ----
          Lcom/android/internal/telephony/DataConnectionTracker$TxRxSum;,
          Lcom/android/internal/telephony/DataConnectionTracker$DataRoamingSettingObserver;,
          Lcom/android/internal/telephony/DataConnectionTracker$Activity;,
!         Lcom/android/internal/telephony/DataConnectionTracker$State;,
!         Lcom/android/internal/telephony/DataConnectionTracker$Injector;
      }
  .end annotation
  
***************
*** 3163,3168 ****
--- 3164,3173 ----
  
      .local v0, result:Z
      :goto_1
+     invoke-static {p0, v0}, Lcom/android/internal/telephony/DataConnectionTracker$Injector;->getAnyDataEnabled(Lcom/android/internal/telephony/DataConnectionTracker;Z)Z
+ 
+     move-result v0
+ 
      monitor-exit v3
      :try_end_0
      .catchall {:try_start_0 .. :try_end_0} :catchall_0
***************
*** 4375,4380 ****
--- 4380,4428 ----
      throw v1
  .end method
  
+ .method protected isMmsDataEnabled()Z
+     .locals 3
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
+     const-string v2, "mms"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     const-string v2, "always_enable_mms"
+ 
+     invoke-static {v1, v2, v0}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
+ 
+     move-result v1
+ 
+     if-ne v1, v0, :cond_0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
+ 
  .method protected abstract log(Ljava/lang/String;)V
  .end method
  
