*** ./smali/com/android/internal/telephony/IccPhoneBookInterfaceManagerProxy.smali	2014-10-11 10:19:58.444207926 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/IccPhoneBookInterfaceManagerProxy.smali	2014-10-11 10:16:59.420204043 +1100
***************
*** 81,86 ****
--- 81,99 ----
      return v0
  .end method
  
+ .method public getAdnCapacity()I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/IccPhoneBookInterfaceManagerProxy;->mIccPhoneBookInterfaceManager:Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;->getAdnCapacity()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
  .method public getAdnRecordsInEf(I)Ljava/util/List;
      .locals 1
      .parameter "efid"
***************
*** 139,144 ****
--- 152,170 ----
  
      move-result v0
  
+     return v0
+ .end method
+ 
+ .method public getFreeAdn()I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/IccPhoneBookInterfaceManagerProxy;->mIccPhoneBookInterfaceManager:Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;->getFreeAdn()I
+ 
+     move-result v0
+ 
      return v0
  .end method
  
