*** ./smali/com/android/internal/telephony/IIccPhoneBook$Stub.smali	2014-10-11 10:20:01.584207994 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/IIccPhoneBook$Stub.smali	2014-10-11 10:17:02.148204102 +1100
***************
*** 28,39 ****
--- 28,43 ----
  
  .field static final TRANSACTION_addContactToGroup:I = 0x11
  
+ .field static final TRANSACTION_getAdnCapacity:I = 0x22
+ 
  .field static final TRANSACTION_getAdnRecordsInEf:I = 0x1
  
  .field static final TRANSACTION_getAdnRecordsSize:I = 0xa
  
  .field static final TRANSACTION_getAnrCount:I = 0x19
  
+ .field static final TRANSACTION_getFreeAdn:I = 0x21
+ 
  .field static final TRANSACTION_getPhonebookMemStorageExt:I = 0x20
  
  .field static final TRANSACTION_getSneRecordLen:I = 0x1f
***************
*** 1509,1514 ****
--- 1513,1564 ----
  
      goto/16 :goto_0
  
+     .end local v14           #_result:[Lcom/android/internal/telephony/gsm/UsimPBMemInfo;
+     :sswitch_21
+     const-string v2, "com.android.internal.telephony.IIccPhoneBook"
+ 
+     move-object/from16 v0, p2
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/IIccPhoneBook$Stub;->getFreeAdn()I
+ 
+     move-result v14
+ 
+     .local v14, _result:I
+     invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     move-object/from16 v0, p3
+ 
+     invoke-virtual {v0, v14}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     const/4 v2, 0x1
+ 
+     goto/16 :goto_0
+ 
+     .end local v14           #_result:I
+     :sswitch_22
+     const-string v2, "com.android.internal.telephony.IIccPhoneBook"
+ 
+     move-object/from16 v0, p2
+ 
+     invoke-virtual {v0, v2}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual/range {p0 .. p0}, Lcom/android/internal/telephony/IIccPhoneBook$Stub;->getAdnCapacity()I
+ 
+     move-result v14
+ 
+     .restart local v14       #_result:I
+     invoke-virtual/range {p3 .. p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     move-object/from16 v0, p3
+ 
+     invoke-virtual {v0, v14}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     const/4 v2, 0x1
+ 
+     goto/16 :goto_0
+ 
      nop
  
      :sswitch_data_0
***************
*** 1545,1550 ****
--- 1595,1602 ----
          0x1e -> :sswitch_1e
          0x1f -> :sswitch_1f
          0x20 -> :sswitch_20
+         0x21 -> :sswitch_21
+         0x22 -> :sswitch_22
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
