*** ./smali/com/android/internal/telephony/IIccPhoneBook.smali	2014-10-11 10:20:01.640207996 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/IIccPhoneBook.smali	2014-10-11 10:17:02.192204103 +1100
***************
*** 23,28 ****
--- 23,36 ----
      .end annotation
  .end method
  
+ .method public abstract getAdnCapacity()I
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
  .method public abstract getAdnRecordsInEf(I)Ljava/util/List;
      .annotation system Ldalvik/annotation/Signature;
          value = {
***************
*** 53,58 ****
--- 61,74 ----
      .annotation system Ldalvik/annotation/Throws;
          value = {
              Landroid/os/RemoteException;
+         }
+     .end annotation
+ .end method
+ 
+ .method public abstract getFreeAdn()I
+     .annotation system Ldalvik/annotation/Throws;
+         value = {
+             Landroid/os/RemoteException;
          }
      .end annotation
  .end method
