*** ./smali/com/android/internal/telephony/SMSDispatcher.smali	2014-10-11 10:20:01.792207999 +1100
--- /home/cosmicdan/patchrom_mtk/bang/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/SMSDispatcher.smali	2014-10-11 10:17:02.340204106 +1100
***************
*** 8,14 ****
      value = {
          Lcom/android/internal/telephony/SMSDispatcher$ConfirmDialogListener;,
          Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;,
!         Lcom/android/internal/telephony/SMSDispatcher$SettingsObserver;
      }
  .end annotation
  
--- 8,15 ----
      value = {
          Lcom/android/internal/telephony/SMSDispatcher$ConfirmDialogListener;,
          Lcom/android/internal/telephony/SMSDispatcher$SmsTracker;,
!         Lcom/android/internal/telephony/SMSDispatcher$SettingsObserver;,
!         Lcom/android/internal/telephony/SMSDispatcher$Injector;
      }
  .end annotation
  
***************
*** 182,188 ****
  
  .field private final mSettingsObserver:Lcom/android/internal/telephony/SMSDispatcher$SettingsObserver;
  
! .field protected mSimId:I
  
  .field protected mSmsCapable:Z
  
--- 183,191 ----
  
  .field private final mSettingsObserver:Lcom/android/internal/telephony/SMSDispatcher$SettingsObserver;
  
! .field protected final mSimId:I
! 
! .field protected final mSlotId:I
  
  .field protected mSmsCapable:Z
  
***************
*** 370,377 ****
  
      iput-boolean v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mStorageAvailable:Z
  
-     iput v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
- 
      iput-boolean v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSmsReady:Z
  
      iput v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->messageCountNeedCopy:I
--- 373,378 ----
***************
*** 604,609 ****
--- 605,622 ----
  
      move-result v3
  
+     iput v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
+ 
+     iget-object v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
+ 
+     iget v6, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
+ 
+     invoke-static {v3, v6}, Landroid/provider/Telephony$SIMInfo;->getIdBySlot(Landroid/content/Context;I)J
+ 
+     move-result-wide v6
+ 
+     long-to-int v3, v6
+ 
      iput v3, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      new-instance v2, Landroid/content/IntentFilter;
***************
*** 678,684 ****
  
      aput-object v7, v6, v5
  
!     iget v7, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-static {v7}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 691,697 ----
  
      aput-object v7, v6, v5
  
!     iget v7, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-static {v7}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 741,747 ****
  
      aput-object v7, v6, v5
  
!     iget v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 754,760 ----
  
      aput-object v7, v6, v5
  
!     iget v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 800,806 ****
  
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
!     iget v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-direct {v3, v4, v5}, Lcom/android/internal/telephony/ConcatenatedSmsFwkExt;-><init>(Landroid/content/Context;I)V
  
--- 813,819 ----
  
      iget-object v4, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
  
!     iget v5, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-direct {v3, v4, v5}, Lcom/android/internal/telephony/ConcatenatedSmsFwkExt;-><init>(Landroid/content/Context;I)V
  
***************
*** 1472,1478 ****
  
      const-string v1, "simId"
  
!     iget v2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
--- 1485,1491 ----
  
      const-string v1, "simId"
  
!     iget v2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-virtual {v0, v1, v2}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
***************
*** 1835,1840 ****
--- 1848,1857 ----
      .parameter "sentIntent"
  
      .prologue
+     const/4 v1, 0x0
+     
+     return v1
+     
      invoke-static {p3}, Lcom/android/internal/telephony/SMSDispatcher;->getAppNameByIntent(Landroid/app/PendingIntent;)Ljava/lang/String;
  
      move-result-object v4
***************
*** 2011,2017 ****
  
      const-string v0, "simId"
  
!     iget v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
--- 2028,2034 ----
  
      const-string v0, "simId"
  
!     iget v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
***************
*** 2051,2057 ****
  
      const-string v0, "simId"
  
!     iget v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
--- 2068,2074 ----
  
      const-string v0, "simId"
  
!     iget v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
  
***************
*** 3348,3354 ****
  
      move-result-object v9
  
!     iget v10, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-virtual {v9, v10}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
--- 3365,3371 ----
  
      move-result-object v9
  
!     iget v10, p0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-virtual {v9, v10}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
***************
*** 4297,4303 ****
  
      move-object/from16 v0, p0
  
!     iget v0, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      move/from16 v20, v0
  
--- 4314,4320 ----
  
      move-object/from16 v0, p0
  
!     iget v0, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      move/from16 v20, v0
  
***************
*** 4652,4658 ****
  
      move-object/from16 v0, p0
  
!     iget v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
--- 4669,4675 ----
  
      move-object/from16 v0, p0
  
!     iget v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
***************
*** 4900,4906 ****
  
      move-object/from16 v0, p0
  
!     iget v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 4917,4923 ----
  
      move-object/from16 v0, p0
  
!     iget v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-static {v3}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
***************
*** 5397,5403 ****
      .local v28, seqNumber:Ljava/lang/String;
      move-object/from16 v0, p0
  
!     iget v4, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-static {v4}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
--- 5414,5420 ----
      .local v28, seqNumber:Ljava/lang/String;
      move-object/from16 v0, p0
  
!     iget v4, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-static {v4}, Ljava/lang/Integer;->toString(I)Ljava/lang/String;
  
***************
*** 5830,5836 ****
  
      move-object/from16 v0, p0
  
!     iget v5, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSimId:I
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
--- 5847,5853 ----
  
      move-object/from16 v0, p0
  
!     iget v5, v0, Lcom/android/internal/telephony/SMSDispatcher;->mSlotId:I
  
      invoke-static {v5}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
  
